
<html>

<head>
	<title>org.molgenis TAB exchange format reference documentation</title>
	
	<meta name="keywords"			content="">
	<meta name="description"		content="">
	<meta http-equiv="Contect-Type"	content="text/html; charset-UTF-8">
	
	<style type="text/css">
		body
		{
			background:		#ffffff;
			color:			#000000;
			font-family:	arial, sans-serif;
			font-size:		10pt;
		}
		
		td
		{
			font-family:	arial, sans-serif;
			font-size:		10pt;
			vertical-align: top;
		}
		.tablehead
		{
			background:		#888888;
			color:			#ffffff;
			text-align:		left;
			font-family:	arial, sans-serif;
			font-size:		10pt;
			font-weight:	bold;
		}
		th
		{
			font-family:	arial, sans-serif;
			font-size:		10pt;
			font-weight:	bold;
		}
	</style>
</head>



<body>
<h1>org.molgenis file format reference</h1>
<p>
This is documentation on the data exchange format for the 'org.molgenis' system. 
</p>
<p>To ease data exchange this system comes with a simple 'tab separated values' file format. 
In such text files the data is formatted in tables with the columns separated using tabs, colons, or semi-colons.
Advantage is that these files can be easily created and parsed using common spreadsheet tools like Excel. 
An example of such tab delimited file is shown below:
<pre>
name	description	date
Experiment1	This is my first experiment	2010-01-19
Experiment2	This is my second experiment	2010-01-20
</pre>
This document describes what file types and columns are defined for the 'org.molgenis' system. 
Data in this format can be uploaded to the database via the user interface using the 'File' menu). 
Alternatively, a whole directory of such files can be loaded in batch using the CsvImport program. 
The following files are currently recognized by this program (grouped by topic):
</p>
<!-- per module -->
<ul>
<li><b></b> files:
<ul>
<li><a href="#molgenisEntity_entity">molgenisentity.txt</a>
<li><a href="#molgenisFile_entity">molgenisfile.txt</a>
<li><a href="#runtimeProperty_entity">runtimeproperty.txt</a>
</ul>
</li>
<li><b></b> files:
<ul>
<li><a href="#molgenisRole_entity">molgenisrole.txt</a>
<li><a href="#molgenisGroup_entity">molgenisgroup.txt</a>
<li><a href="#molgenisUser_entity">molgenisuser.txt</a>
<li><a href="#molgenisRoleGroupLink_entity">molgenisrolegrouplink.txt</a>
<li><a href="#molgenisPermission_entity">molgenispermission.txt</a>
</ul>
</li>
<li><b></b> files:
<ul>
<li><a href="#characteristic_entity">characteristic.txt</a>
<li><a href="#observationTarget_entity">observationtarget.txt</a>
<li><a href="#observableFeature_entity">observablefeature.txt</a>
<li><a href="#category_entity">category.txt</a>
<li><a href="#protocol_entity">protocol.txt</a>
<li><a href="#dataSet_entity">dataset.txt</a>
<li><a href="#observationSet_entity">observationset.txt</a>
<li><a href="#observedValue_entity">observedvalue.txt</a>
</ul>
</li>
<li><b></b> files:
<ul>
<li><a href="#species_entity">species.txt</a>
<li><a href="#individual_entity">individual.txt</a>
<li><a href="#panel_entity">panel.txt</a>
<li><a href="#panelSource_entity">panelsource.txt</a>
<li><a href="#ontology_entity">ontology.txt</a>
<li><a href="#ontologyTerm_entity">ontologyterm.txt</a>
<li><a href="#accession_entity">accession.txt</a>
</ul>
</li>
<li><b></b> files:
<ul>
<li><a href="#genome_entity">genome.txt</a>
<li><a href="#chromosome_entity">chromosome.txt</a>
<li><a href="#gene_entity">gene.txt</a>
<li><a href="#protein_entity">protein.txt</a>
<li><a href="#proteinDomain_entity">proteindomain.txt</a>
<li><a href="#exon_entity">exon.txt</a>
<li><a href="#variant_entity">variant.txt</a>
</ul>
</li>
<li><b></b> files:
<ul>
<li><a href="#study_entity">study.txt</a>
<li><a href="#experiment_entity">experiment.txt</a>
<li><a href="#institute_entity">institute.txt</a>
<li><a href="#person_entity">person.txt</a>
<li><a href="#citation_entity">citation.txt</a>
<li><a href="#contribution_entity">contribution.txt</a>
<li><a href="#submission_entity">submission.txt</a>
</ul>
</li>
<li><b></b> files:
<ul>
<li><a href="#investigation_entity">investigation.txt</a>
<li><a href="#studyDetails_entity">studydetails.txt</a>
<li><a href="#frequencyCluster_entity">frequencycluster.txt</a>
<li><a href="#genotypeFrequency_entity">genotypefrequency.txt</a>
<li><a href="#alleleFrequency_entity">allelefrequency.txt</a>
<li><a href="#phenotypeProperty_entity">phenotypeproperty.txt</a>
<li><a href="#phenotypeMethod_entity">phenotypemethod.txt</a>
<li><a href="#phenotypeValue_entity">phenotypevalue.txt</a>
<li><a href="#samplePanel_entity">samplepanel.txt</a>
<li><a href="#assayedPanel_entity">assayedpanel.txt</a>
<li><a href="#gWASExperiment_entity">gwasexperiment.txt</a>
<li><a href="#usedMarkerSet_entity">usedmarkerset.txt</a>
<li><a href="#significance_entity">significance.txt</a>
<li><a href="#effectSize_entity">effectsize.txt</a>
<li><a href="#selectionCriteria_entity">selectioncriteria.txt</a>
</ul>
</li>
<!-- outside module -->
<li><b>org.molgenis</b> files:
<ul>
<li><a href="#molgenisEntity_entity">molgenisentity.txt</a>
<li><a href="#molgenisFile_entity">molgenisfile.txt</a>
<li><a href="#runtimeProperty_entity">runtimeproperty.txt</a>
<li><a href="#molgenisRole_entity">molgenisrole.txt</a>
<li><a href="#molgenisGroup_entity">molgenisgroup.txt</a>
<li><a href="#molgenisUser_entity">molgenisuser.txt</a>
<li><a href="#molgenisRoleGroupLink_entity">molgenisrolegrouplink.txt</a>
<li><a href="#molgenisPermission_entity">molgenispermission.txt</a>
<li><a href="#characteristic_entity">characteristic.txt</a>
<li><a href="#observationTarget_entity">observationtarget.txt</a>
<li><a href="#observableFeature_entity">observablefeature.txt</a>
<li><a href="#category_entity">category.txt</a>
<li><a href="#protocol_entity">protocol.txt</a>
<li><a href="#dataSet_entity">dataset.txt</a>
<li><a href="#observationSet_entity">observationset.txt</a>
<li><a href="#observedValue_entity">observedvalue.txt</a>
<li><a href="#species_entity">species.txt</a>
<li><a href="#individual_entity">individual.txt</a>
<li><a href="#panel_entity">panel.txt</a>
<li><a href="#panelSource_entity">panelsource.txt</a>
<li><a href="#ontology_entity">ontology.txt</a>
<li><a href="#ontologyTerm_entity">ontologyterm.txt</a>
<li><a href="#accession_entity">accession.txt</a>
<li><a href="#genome_entity">genome.txt</a>
<li><a href="#chromosome_entity">chromosome.txt</a>
<li><a href="#gene_entity">gene.txt</a>
<li><a href="#protein_entity">protein.txt</a>
<li><a href="#proteinDomain_entity">proteindomain.txt</a>
<li><a href="#exon_entity">exon.txt</a>
<li><a href="#variant_entity">variant.txt</a>
<li><a href="#study_entity">study.txt</a>
<li><a href="#experiment_entity">experiment.txt</a>
<li><a href="#institute_entity">institute.txt</a>
<li><a href="#person_entity">person.txt</a>
<li><a href="#citation_entity">citation.txt</a>
<li><a href="#contribution_entity">contribution.txt</a>
<li><a href="#submission_entity">submission.txt</a>
<li><a href="#investigation_entity">investigation.txt</a>
<li><a href="#studyDetails_entity">studydetails.txt</a>
<li><a href="#frequencyCluster_entity">frequencycluster.txt</a>
<li><a href="#genotypeFrequency_entity">genotypefrequency.txt</a>
<li><a href="#alleleFrequency_entity">allelefrequency.txt</a>
<li><a href="#phenotypeProperty_entity">phenotypeproperty.txt</a>
<li><a href="#phenotypeMethod_entity">phenotypemethod.txt</a>
<li><a href="#phenotypeValue_entity">phenotypevalue.txt</a>
<li><a href="#samplePanel_entity">samplepanel.txt</a>
<li><a href="#assayedPanel_entity">assayedpanel.txt</a>
<li><a href="#gWASExperiment_entity">gwasexperiment.txt</a>
<li><a href="#usedMarkerSet_entity">usedmarkerset.txt</a>
<li><a href="#significance_entity">significance.txt</a>
<li><a href="#effectSize_entity">effectsize.txt</a>
<li><a href="#selectionCriteria_entity">selectioncriteria.txt</a>
</ul>
</li>
</ul>
Below, the columns for each of these file types are detailed as well as example data shown (if available).


<!-- entities inside modules -->
<h2>org.molgenis.core file types</h2>

<h3 id="molgenisEntity_entity">File: molgenisentity.txt</h3> 
<p>Contents:<br> Referenceable catalog of entity names, menus, forms and
				plugins.
			
</p>
<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Name of the entity.<br></td>
	</tr>
	<tr>
		<td style="width:150px">type_</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Type of the entity.<br></td>
	</tr>
	<tr>
		<td style="width:150px">classname</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Full name of the entity.<br></td>
	</tr>
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column classname should unique.</td></tr>
<tr><td colspan="5">Contraint: values in the combined columns (name, type_) should be unique.</td></tr>
</table>
</p>


<h3 id="molgenisFile_entity">File: molgenisfile.txt</h3> 
<p>Contents:<br> 
				Helper entity to deal with files. Has a decorator to regulate
				storage and coupling to an Entity. Do not make abstract because of
				subtyping. This means the names of the subclasses will be used to
				distinguish MolgenisFiles and place them in the correct folders.
			
</p>
<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">identifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>user supplied or automatically assigned (using a decorator) unique and short identifier, e.g. MA1234.<br></td>
	</tr>
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>human readible name, not necessary unique.<br></td>
	</tr>
	<tr>
		<td style="width:150px">extension</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>The file extension. This will be mapped to MIME type at runtime. For example, a type 'png' will be served out as 'image/png'.<br></td>
	</tr>
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column identifier should unique.</td></tr>
<tr><td colspan="5">Constraint: values in column name should unique.</td></tr>
</table>
</p>


<h3 id="runtimeProperty_entity">File: runtimeproperty.txt</h3> 

<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">identifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>user supplied or automatically assigned (using a decorator) unique and short identifier, e.g. MA1234.<br></td>
	</tr>
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>human readible name, not necessary unique.<br></td>
	</tr>
	<tr>
		<td style="width:150px">value</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Value.<br></td>
	</tr>
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column identifier should unique.</td></tr>
<tr><td colspan="5">Constraint: values in column name should unique.</td></tr>
</table>
</p>


<h2>org.molgenis.auth file types</h2>

<h3 id="molgenisRole_entity">File: molgenisrole.txt</h3> 

<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>name.<br></td>
	</tr>
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column name should unique.</td></tr>
</table>
</p>


<h3 id="molgenisGroup_entity">File: molgenisgroup.txt</h3> 

<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>name.<br></td>
	</tr>
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column name should unique.</td></tr>
</table>
</p>


<h3 id="molgenisUser_entity">File: molgenisuser.txt</h3> 
<p>Contents:<br> Anyone who can login
.</p>
<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">username</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>username.<br></td>
	</tr>
	<tr>
		<td style="width:150px">password_</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">secret</td>
		<td>big fixme: password type.<br></td>
	</tr>
	<tr>
		<td style="width:150px">activationcode</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Used as alternative authentication mechanism to verify user email and/or if user has lost password.<br></td>
	</tr>
	<tr>
		<td style="width:150px">active</td>
		<td style="width:50px">bool</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">false</td>
		<td>Boolean to indicate if this account can be used to login.<br></td>
	</tr>
	<tr>
		<td style="width:150px">superuser</td>
		<td style="width:50px">bool</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">false</td>
		<td>superuser.<br></td>
	</tr>
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column username should unique.</td></tr>
</table>
</p>


<h3 id="molgenisRoleGroupLink_entity">File: molgenisrolegrouplink.txt</h3> 

<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">identifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>user supplied or automatically assigned (using a decorator) unique and short identifier, e.g. MA1234.<br></td>
	</tr>
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>human readible name, not necessary unique.<br></td>
	</tr>
	<tr>
		<td style="width:150px">group__name<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>group_.
		This xref uses {group__name} to find related elements in file <a href="#molgenisGroup_entity">molgenisGroup.txt</a> based on unique column {name}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">role__name<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>role_.
		This xref uses {role__name} to find related elements in file <a href="#molgenisRole_entity">molgenisRole.txt</a> based on unique column {name}.
		</td>
	</tr>	
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column identifier should unique.</td></tr>
</table>
</p>


<h3 id="molgenisPermission_entity">File: molgenispermission.txt</h3> 

<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">role__name<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>role_.
		This xref uses {role__name} to find related elements in file <a href="#molgenisRole_entity">molgenisRole.txt</a> based on unique column {name}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">entity_className<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>entity.
		This xref uses {entity_classname} to find related elements in file <a href="#molgenisEntity_entity">molgenisEntity.txt</a> based on unique column {classname}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">permission</td>
		<td style="width:50px">enum</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>permission.<br></td>
	</tr>
	<!-- all optional fields -->
	<!-- all the uniques -->
</table>
</p>


<h2>org.molgenis.observ file types</h2>
<p>
			Observ-OM is a model to uniformly describe any phenotypic, genotypic
			or
			molecular observation. The four core concepts are:
			<ul>
        <li>
            <b>ObservationTarget:</b>
					the object we are observing.
					<br/>
					For example: an Individual,
					a Panel/Strain, a Sample, a Gene, a
					Marker.
				</li>
        <li>
            <b>ObservableFeature:</b>
					the question asked or characteristic being
					oberveded.
					<br/>
					For example: 'What is height of [Target]?', 'What is genotype of
					[Target] at
					[Marker]?', 'Has [Target] ever used medicine [X]', 'What
					is QTL p-value
					association of [Marker] with [Gene Expression]?
				</li>
        <li>
            <b>Protocol:</b>
					the procedure of observation
					<br/>
					For example: Food Frequency Questionnaire (FFQ), R/QTL mapping
					method,
					Genotype array, GWAS protocol.
				</li>
        <li>
            <b>Observation:</b>
					the actual act of observation. For example:
					'FFQ asked to
					[Individual:individual1], 'R/QTL applied to
					[Marker:pvv4]',
					'Genotype array applied to [Panel:Mouse RIL BxD 1].
					Each observation
					can have multiple ObservedValue AND be grouped into a DataSet.
				</li>
    </ul>

</p>
<h3 id="characteristic_entity">File: characteristic.txt</h3> 
<p>Contents:<br>  Characteristics are yes-no
				statements about things in
				the world. These can be used as part of an observation, as
				parameter
				of ObservableFeature ('measuredCharacteristic'). For
				example: 'What
				is allele of [Marker]', here the [Marker] is a
				characteristic. Also,
				Characteristics can be used as target of
				observation. Typical
				examples are 'Individual' or 'Panel'. But also
				'Marker' can be an
				Target when asked the question 'QTL p-value for
				[phenotype]': here
				both target and feature are characteristic, for
				example 'leave count'
				(phenotype characteristic) and 'PVV4' (marker
				characteristic).
			
</p>
<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">identifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>user supplied or automatically assigned (using a decorator) unique and short identifier, e.g. MA1234.<br></td>
	</tr>
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>human readible name, not necessary unique.<br></td>
	</tr>
	<tr>
		<td style="width:150px">description</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>(Optional) Rudimentary meta data about the observable feature. Use of ontology       terms references to establish unambigious descriptions is recommended.<br></td>
	</tr>
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column identifier should unique.</td></tr>
</table>
</p>


<h3 id="observationTarget_entity">File: observationtarget.txt</h3> 
<p>Contents:<br>  ObservationTarget defines subjects of
				observation, such
				as Individual, Panel, Sample, etc. For instance: 'target 1' IS A
				'Individual'.
			
</p>
<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">identifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>user supplied or automatically assigned (using a decorator) unique and short identifier, e.g. MA1234.<br></td>
	</tr>
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>human readible name, not necessary unique.<br></td>
	</tr>
	<tr>
		<td style="width:150px">description</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>(Optional) Rudimentary meta data about the observable feature. Use of ontology       terms references to establish unambigious descriptions is recommended.<br></td>
	</tr>
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column identifier should unique.</td></tr>
</table>
</p>


<h3 id="observableFeature_entity">File: observablefeature.txt</h3> 
<p>Contents:<br> 
				ObservableFeature defines anything that can be observed.
				<p/>
				In other words, ObservableFeature are the
				questions asked, e.g. 'What
				is Height?', 'What is Systolic blood
				pressure?', or 'Has blue eyes?'.
				<p/>
				Some questions may
				be repeated for multiple characteristics. For
				example 'What is [MarkerAllele]
				observed?' can be applied to all
				elements of a MarkerSet, and 'What
				is [medicin codes] uses' can be
				applied to a set of Medicine codes. This can be specified using the
				measuredCharacteristic field.
				<p/>
				The identifier of ObservableFeature is globally
				unique. It is
				recommended that
				each
				ObservableFeature is named
				according to a
				well-defined ontology
				term or database accession.
			
</p>
<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">identifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>user supplied or automatically assigned (using a decorator) unique and short identifier, e.g. MA1234.<br></td>
	</tr>
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>human readible name, not necessary unique.<br></td>
	</tr>
	<tr>
		<td style="width:150px">description</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>(Optional) Rudimentary meta data about the observable feature. Use of ontology       terms references to establish unambigious descriptions is recommended.<br></td>
	</tr>
	<tr>
		<td style="width:150px">unit_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>(Optional) Reference to the well-defined measurement unit used to observe this feature       (if feature is that concrete). E.g. mmHg.
		This xref uses {unit_identifier} to find related elements in file <a href="#ontologyTerm_entity">ontologyTerm.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">datatype</td>
		<td style="width:50px">enum</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">string</td>
		<td>(Optional) Reference to the technical data type. E.g. 'int'.<br></td>
	</tr>
	<tr>
		<td style="width:150px">temporal</td>
		<td style="width:50px">bool</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">false</td>
		<td>Whether this feature is time dependent and can have different values when measured       on different times (e.g. weight, temporal=true) or generally only measured once (e.g. birth date,       temporal=false).<br></td>
	</tr>
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column identifier should unique.</td></tr>
</table>
</p>


<h3 id="category_entity">File: category.txt</h3> 
<p>Contents:<br> Category is partOf ObservableFeature to define
				categories for an
				ObservableFeature, such as the categorical
				answer codes that are often used in Questionaires. 
				For example the ObservableFeature 'sex' has
				{code_string = 1, label=male} and {code_string
				= 2, label=female}.
				Category can be linked to well-defined
				ontology terms via the
				ontologyReference. Category
				extends
				ObservationElement such that it
				can be referenced by
				ObservedValue.value.
				The Category class maps to
				METABASE::Category
			
.</p>
<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">identifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>user supplied or automatically assigned (using a decorator) unique and short identifier, e.g. MA1234.<br></td>
	</tr>
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>human readible name, not necessary unique.<br></td>
	</tr>
	<tr>
		<td style="width:150px">description</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>(Optional) Rudimentary meta data about the observable feature. Use of ontology       terms references to establish unambigious descriptions is recommended.<br></td>
	</tr>
	<tr>
		<td style="width:150px">observablefeature_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>The Measurement these permitted values are part of..
		This xref uses {observablefeature_identifier} to find related elements in file <a href="#observableFeature_entity">observableFeature.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">valuecode</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>The value used to store this category in ObservedValue. For example '1', '2'.<br></td>
	</tr>
	<tr>
		<td style="width:150px">ismissing</td>
		<td style="width:50px">bool</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">false</td>
		<td>whether this value should be treated as missing value.<br></td>
	</tr>
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column identifier should unique.</td></tr>
</table>
</p>


<h3 id="protocol_entity">File: protocol.txt</h3> 
<p>Contents:<br> 
				The Protocol class defines parameterizable descriptions
				of
				(analysis)methods. Examples of protocols are: Questionaires, SOPs,
				Assay platforms, Statistical analyses, etc.
				Each protocol has a
				unique identifier.
				Protocol has an association to OntologyTerm to
				represent the type of
				protocol.
			
</p>
<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">identifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>user supplied or automatically assigned (using a decorator) unique and short identifier, e.g. MA1234.<br></td>
	</tr>
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>human readible name, not necessary unique.<br></td>
	</tr>
	<tr>
		<td style="width:150px">description</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>(Optional) Rudimentary meta data about the observable feature. Use of ontology       terms references to establish unambigious descriptions is recommended.<br></td>
	</tr>
	<tr>
		<td style="width:150px">protocoltype_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>classification of protocol.
		This xref uses {protocoltype_identifier} to find related elements in file <a href="#ontologyTerm_entity">ontologyTerm.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">subprotocols_Identifier<br></td>
		<td style="width:50px">mref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Subprotocols of this protocol.
		This mref uses {subprotocols_identifier} to find related elements in file <a href="#protocol_entity">protocol.txt</a> based on unique column {identifier}.
		. More than one reference can be added separated by '|', for example: ref1|ref2|ref3.</td>
	</tr>	
	<tr>
		<td style="width:150px">features_Identifier<br></td>
		<td style="width:50px">mref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>parameters (in/out) that are used or produced by this protocol..
		This mref uses {features_identifier} to find related elements in file <a href="#observableFeature_entity">observableFeature.txt</a> based on unique column {identifier}.
		. More than one reference can be added separated by '|', for example: ref1|ref2|ref3.</td>
	</tr>	
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column identifier should unique.</td></tr>
</table>
</p>


<h3 id="dataSet_entity">File: dataset.txt</h3> 
<p>Contents:<br> Container for one or more observations that are measured
				using the same protocol and by the same performer(s). The dataset
				may be a file (having the same
				identifier) but in most cases it is a
				data table consisting of rows (Observation).

				This entity replaces
				ProtocolApplication.
			
</p>
<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">identifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>user supplied or automatically assigned (using a decorator) unique and short identifier, e.g. MA1234.<br></td>
	</tr>
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>human readible name, not necessary unique.<br></td>
	</tr>
	<tr>
		<td style="width:150px">description</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>(Optional) Rudimentary meta data about the observable feature. Use of ontology       terms references to establish unambigious descriptions is recommended.<br></td>
	</tr>
	<tr>
		<td style="width:150px">protocolused_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Reference to the protocol that is being used (if available).
		This xref uses {protocolused_identifier} to find related elements in file <a href="#protocol_entity">protocol.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">starttime</td>
		<td style="width:50px">datetime</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">today</td>
		<td>time when the protocol started.<br></td>
	</tr>
	<tr>
		<td style="width:150px">endtime</td>
		<td style="width:50px">datetime</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">today</td>
		<td>(Optional) time when the protocol ended.<br></td>
	</tr>
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column identifier should unique.</td></tr>
</table>
</p>


<h3 id="observationSet_entity">File: observationset.txt</h3> 
<p>Contents:<br> In practice: Observation is one row within a DataSet.
			
</p>
<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">partofdataset_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>DataSet this ValueSet is part of..
		This xref uses {partofdataset_identifier} to find related elements in file <a href="#dataSet_entity">dataSet.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">target_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>References the target for which this data was recorded. For example 'individual1'..
		This xref uses {target_identifier} to find related elements in file <a href="#characteristic_entity">characteristic.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">time</td>
		<td style="width:50px">datetime</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Time of this observationSet.<br></td>
	</tr>
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Contraint: values in the combined columns (partofdataset, target, time) should be unique.</td></tr>
</table>
</p>


<h3 id="observedValue_entity">File: observedvalue.txt</h3> 
<p>Contents:<br> 
				Generic storage of values as part of one observation
				event. Values are atomatic observations,
				e.g.,
				length (feature) of
				individual 1 (valueset.target)
				= 179cm (value).
				Values can also be
				qualified by some characteristic,
				e.g., QTL
				p-value (feature)
				between
				phenotype 'leaf count'
				(characteristic) and
				marker 'PVV4'
				(valueset.target) =
				0.1^10+3 (value).
			
</p>
<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">observationset_id<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Reference to the observation. For example a particular patient visit or the application of a microarray or the calculation of a QTL model.
		This xref uses {observationset_id} to find related elements in file <a href="#observationSet_entity">observationSet.txt</a> based on unique column {id}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">feature_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>References the ObservableFeature that this observation was made on. For example 'probe123'..
		This xref uses {feature_identifier} to find related elements in file <a href="#observableFeature_entity">observableFeature.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">characteristic_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Is brother of [characteristic], or Average of [height].
		This xref uses {characteristic_identifier} to find related elements in file <a href="#characteristic_entity">characteristic.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">value</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>The value observed.<br></td>
	</tr>
	<!-- all optional fields -->
	<!-- all the uniques -->
</table>
</p>


<h2>org.molgenis.observ.target file types</h2>

<h3 id="species_entity">File: species.txt</h3> 
<p>Contents:<br> Ontology terms for species. E.g. Arabidopsis thaliana.
				DISCUSSION: should we avoid subclasses of OntologyTerm and instead
				make a 'tag' filter on terms so we can make pulldowns context
				dependent (e.g. to only show particular subqueries of ontologies).
			
</p>
<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">identifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>user supplied or automatically assigned (using a decorator) unique and short identifier, e.g. MA1234.<br></td>
	</tr>
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>human readible name, not necessary unique.<br></td>
	</tr>
	<tr>
		<td style="width:150px">ontology_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>(Optional) The source ontology or controlled vocabulary list that ontology terms have been obtained from..
		This xref uses {ontology_identifier} to find related elements in file <a href="#ontology_entity">ontology.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">termaccession</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>(Optional) The accession number assigned to the ontology term in its source ontology. If empty it is assumed to be a locally defined term.<br></td>
	</tr>
	<tr>
		<td style="width:150px">definition</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>(Optional) The definition of the term.<br></td>
	</tr>
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column identifier should unique.</td></tr>
<tr><td colspan="5">Contraint: values in the combined columns (ontology, termaccession) should be unique.</td></tr>
</table>
</p>


<h3 id="individual_entity">File: individual.txt</h3> 
<p>Contents:<br>  The Individuals class defines the subjects that are
				used
				as observation target. The Individual class maps to
				XGAP:Individual
				and PaGE:Individual. Groups of individuals can be
				defined via
				Panel.
			
</p>
<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">identifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>user supplied or automatically assigned (using a decorator) unique and short identifier, e.g. MA1234.<br></td>
	</tr>
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>human readible name, not necessary unique.<br></td>
	</tr>
	<tr>
		<td style="width:150px">description</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>(Optional) Rudimentary meta data about the observable feature. Use of ontology       terms references to establish unambigious descriptions is recommended.<br></td>
	</tr>
	<tr>
		<td style="width:150px">mother_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Refers to the mother of the individual..
		This xref uses {mother_identifier} to find related elements in file <a href="#individual_entity">individual.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">father_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Refers to the father of the individual..
		This xref uses {father_identifier} to find related elements in file <a href="#individual_entity">individual.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column identifier should unique.</td></tr>
</table>
</p>


<h3 id="panel_entity">File: panel.txt</h3> 
<p>Contents:<br> The Panel class defines groups of individuals based on
				cohort design, case/controls, families, etc. For instance:
				'LifeLines
				cohort', 'middle aged man', 'recombinant mouse inbred Line
				dba x b6'
				or 'Smith family'. A Panel can act as a single
				ObservationTarget.
				For example: average height (Measurement) in the
				LifeLines cohort
				(Panel) is 174cm (ObservedValue). The Panel class
				maps to XGAP:Strain and PaGE:Panel
				classes. In METABASE this is
				assumed there is one panel per study.
			
</p>
<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">identifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>user supplied or automatically assigned (using a decorator) unique and short identifier, e.g. MA1234.<br></td>
	</tr>
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>human readible name, not necessary unique.<br></td>
	</tr>
	<tr>
		<td style="width:150px">description</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>(Optional) Rudimentary meta data about the observable feature. Use of ontology       terms references to establish unambigious descriptions is recommended.<br></td>
	</tr>
	<tr>
		<td style="width:150px">paneltype_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Indicate the type of Panel (example: Sample panel, AssayedPanel, Natural=wild type, Parental=parents of a cross, F1=First generation of cross, RCC=Recombinant congenic, CSS=chromosome substitution).
		This xref uses {paneltype_identifier} to find related elements in file <a href="#ontologyTerm_entity">ontologyTerm.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">numberofindividuals</td>
		<td style="width:50px">int</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>NumberOfIndividuals.<br></td>
	</tr>
	<tr>
		<td style="width:150px">species_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>The species this panel is an instance of/part of/extracted from..
		This xref uses {species_identifier} to find related elements in file <a href="#species_entity">species.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">individuals_Identifier<br></td>
		<td style="width:50px">mref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>The list of individuals in this panel.
		This mref uses {individuals_identifier} to find related elements in file <a href="#individual_entity">individual.txt</a> based on unique column {identifier}.
		. More than one reference can be added separated by '|', for example: ref1|ref2|ref3.</td>
	</tr>	
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column identifier should unique.</td></tr>
</table>
</p>


<h3 id="panelSource_entity">File: panelsource.txt</h3> 
<p>Contents:<br> PanelSources is partOf Panel to define how panels are
				related panels, founder panels,
				such as overlap,
				selection criteria,
				getting assayed panel from a
				sample panel, etc.
			
</p>
<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">currentpanel_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Panel for which these sources are defined..
		This xref uses {currentpanel_identifier} to find related elements in file <a href="#panel_entity">panel.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">sourcepanel_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Source that contributed individuals to current panel.
		This xref uses {sourcepanel_identifier} to find related elements in file <a href="#panel_entity">panel.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">numberofindividuals</td>
		<td style="width:50px">int</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Number of individuals lifted over from this source.<br></td>
	</tr>
	<tr>
		<td style="width:150px">selectioncriteria</td>
		<td style="width:50px">text</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Inclusion/exclusion criteria used to select these individuals from source into current panel.<br></td>
	</tr>
	<!-- all optional fields -->
	<!-- all the uniques -->
</table>
</p>


<h3 id="ontology_entity">File: ontology.txt</h3> 
<p>Contents:<br>  Ontology defines a reference to an ontology or
				controlled vocabulary from which well-defined and stable (ontology)
				terms can be obtained. Each Ontology should have a unique identifer,
				for instance: Gene Ontology, Mammalian Phenotype, Human Phenotype
				Ontology, Unified Medical Language System, Medical Subject Headings,
				etc. Also a abbreviation is required, for instance: GO, MP, HPO,
				UMLS, MeSH, etc. Use of existing ontologies/vocabularies is
				recommended to harmonize phenotypic feature and value descriptions.
				But one can also create a 'local' Ontology. The Ontology class maps
				to FuGE::Ontology, MAGE-TAB::TermSourceREF.
			
</p>
<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">identifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>user supplied or automatically assigned (using a decorator) unique and short identifier, e.g. MA1234.<br></td>
	</tr>
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>human readible name, not necessary unique.<br></td>
	</tr>
	<tr>
		<td style="width:150px">ontologyaccession</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>A accession that uniquely identifies the ontology (typically an acronym). E.g. GO, MeSH, HPO.<br></td>
	</tr>
	<tr>
		<td style="width:150px">ontologyuri</td>
		<td style="width:50px">hyperlink</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>(Optional) A URI that references the location of the ontology.<br></td>
	</tr>
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column identifier should unique.</td></tr>
</table>
</p>


<h3 id="ontologyTerm_entity">File: ontologyterm.txt</h3> 
<p>Contents:<br> 
				OntologyTerm defines a single entry (term) from an
				ontology or a controlled vocabulary (defined by Ontology). The
				identifier is the ontology term is unique. E.g. 'NCI:Antigen Gene'.
				Other data entities can reference to this OntologyTerm to harmonize
				naming of concepts. If no suitable ontology term exists then one can
				define new terms locally (in which case there is no formal accession
				for the term limiting its use for cross-Investigation queries).
			
</p>
<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">identifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>user supplied or automatically assigned (using a decorator) unique and short identifier, e.g. MA1234.<br></td>
	</tr>
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>human readible name, not necessary unique.<br></td>
	</tr>
	<tr>
		<td style="width:150px">ontology_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>(Optional) The source ontology or controlled vocabulary list that ontology terms have been obtained from..
		This xref uses {ontology_identifier} to find related elements in file <a href="#ontology_entity">ontology.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">termaccession</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>(Optional) The accession number assigned to the ontology term in its source ontology. If empty it is assumed to be a locally defined term.<br></td>
	</tr>
	<tr>
		<td style="width:150px">definition</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>(Optional) The definition of the term.<br></td>
	</tr>
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column identifier should unique.</td></tr>
<tr><td colspan="5">Contraint: values in the combined columns (ontology, termaccession) should be unique.</td></tr>
</table>
</p>


<h3 id="accession_entity">File: accession.txt</h3> 
<p>Contents:<br> 
				An external identifier for an annotation. For example:
				name='R13H8.1', ontology='ensembl' or name='WBgene00000912',
				ontology='wormbase'.
			
</p>
<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">identifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>user supplied or automatically assigned (using a decorator) unique and short identifier, e.g. MA1234.<br></td>
	</tr>
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>human readible name, not necessary unique.<br></td>
	</tr>
	<tr>
		<td style="width:150px">ontology_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>(Optional) The source ontology or controlled vocabulary list that ontology terms have been obtained from..
		This xref uses {ontology_identifier} to find related elements in file <a href="#ontology_entity">ontology.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">termaccession</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>(Optional) The accession number assigned to the ontology term in its source ontology. If empty it is assumed to be a locally defined term.<br></td>
	</tr>
	<tr>
		<td style="width:150px">definition</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>(Optional) The definition of the term.<br></td>
	</tr>
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column identifier should unique.</td></tr>
<tr><td colspan="5">Contraint: values in the combined columns (ontology, termaccession) should be unique.</td></tr>
</table>
</p>


<h2>org.molgenis.variant file types</h2>
<p>
			Schema for using genomic elements such as gene, chromosome, protein
			as extension on 'observ.xml'.
			Generic schema inspired by Chado. All
			other properties should already be
			covered in the core Pheno.xml
			model, in particular relationship which
			is in ObservedValue. Some
			extra may need to be added (e.g. dbXrefs,
			pubs, etc, for each
			ObservationElement).
			<br/>
			There is a crucial difference between the Chado location model and
			the sequence location model used in other schemas, such as GFF,
			GenBank, BioSQL, or BioPerl. First, Chado is the only model to use
			the concept of rank and locgroup. Second, and perhaps more important,
			all these other models allow discontiguous locations (also known as
			"split locations").See
			http://gmod.org/wiki/Chado_Sequence_Module#Table:_feature
		
</p>
<h3 id="genome_entity">File: genome.txt</h3> 

<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">identifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>user supplied or automatically assigned (using a decorator) unique and short identifier, e.g. MA1234.<br></td>
	</tr>
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>human readible name, not necessary unique.<br></td>
	</tr>
	<tr>
		<td style="width:150px">description</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>(Optional) Rudimentary meta data about the observable feature. Use of ontology       terms references to establish unambigious descriptions is recommended.<br></td>
	</tr>
	<tr>
		<td style="width:150px">residues</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>A sequence of alphabetic characters representing biological residues (nucleic acids, amino acids). This column does not need to be manifested for all features; it is optional for features such as exons where the residues can be derived from the featureloc. It is recommended that the value for this column be manifested for features which may may non-contiguous sublocations (e.g. transcripts), since derivation at query time is non-trivial. For expressed sequence, the DNA sequence should be used rather than the RNA sequence.<br></td>
	</tr>
	<tr>
		<td style="width:150px">seqlen</td>
		<td style="width:50px">int</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>The length of the residue feature. See column:residues. This column is partially redundant with the residues column, and also with featureloc. This column is required because the location may be unknown and the residue sequence may not be manifested, yet it may be desirable to store and query the length of the feature. The seqlen should always be manifested where the length of the sequence is known.<br></td>
	</tr>
	<tr>
		<td style="width:150px">species_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>species this genome belongs to..
		This xref uses {species_identifier} to find related elements in file <a href="#species_entity">species.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column identifier should unique.</td></tr>
</table>
</p>


<h3 id="chromosome_entity">File: chromosome.txt</h3> 
<p>Contents:<br> Example: b37:chr1.
</p>
<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">identifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>user supplied or automatically assigned (using a decorator) unique and short identifier, e.g. MA1234.<br></td>
	</tr>
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>human readible name, not necessary unique.<br></td>
	</tr>
	<tr>
		<td style="width:150px">description</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>(Optional) Rudimentary meta data about the observable feature. Use of ontology       terms references to establish unambigious descriptions is recommended.<br></td>
	</tr>
	<tr>
		<td style="width:150px">residues</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>A sequence of alphabetic characters representing biological residues (nucleic acids, amino acids). This column does not need to be manifested for all features; it is optional for features such as exons where the residues can be derived from the featureloc. It is recommended that the value for this column be manifested for features which may may non-contiguous sublocations (e.g. transcripts), since derivation at query time is non-trivial. For expressed sequence, the DNA sequence should be used rather than the RNA sequence.<br></td>
	</tr>
	<tr>
		<td style="width:150px">seqlen</td>
		<td style="width:50px">int</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>The length of the residue feature. See column:residues. This column is partially redundant with the residues column, and also with featureloc. This column is required because the location may be unknown and the residue sequence may not be manifested, yet it may be desirable to store and query the length of the feature. The seqlen should always be manifested where the length of the sequence is known.<br></td>
	</tr>
	<tr>
		<td style="width:150px">genome_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>genome.
		This xref uses {genome_identifier} to find related elements in file <a href="#genome_entity">genome.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">ordernr</td>
		<td style="width:50px">int</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>orderNr.<br></td>
	</tr>
	<tr>
		<td style="width:150px">isautosomal</td>
		<td style="width:50px">bool</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Is 'yes' when number of chromosomes is equal in male and female individuals, i.e., if not a sex chromosome.<br></td>
	</tr>
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column identifier should unique.</td></tr>
</table>
</p>


<h3 id="gene_entity">File: gene.txt</h3> 
<p>Contents:<br> Serves as a view on SequenceCharacteristics that are
				genes / cdna
			
.</p>
<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">identifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>user supplied or automatically assigned (using a decorator) unique and short identifier, e.g. MA1234.<br></td>
	</tr>
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>human readible name, not necessary unique.<br></td>
	</tr>
	<tr>
		<td style="width:150px">description</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>(Optional) Rudimentary meta data about the observable feature. Use of ontology       terms references to establish unambigious descriptions is recommended.<br></td>
	</tr>
	<tr>
		<td style="width:150px">gdna_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>The genome this element lies on..
		This xref uses {gdna_identifier} to find related elements in file <a href="#chromosome_entity">chromosome.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">gdna_start</td>
		<td style="width:50px">int</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Start position on genomic sequence.<br></td>
	</tr>
	<tr>
		<td style="width:150px">gdna_end</td>
		<td style="width:50px">int</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>End position on genomic sequence.<br></td>
	</tr>
	<tr>
		<td style="width:150px">residues</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>A sequence of alphabetic characters representing biological residues (nucleic acids, amino acids). This column does not need to be manifested for all features; it is optional for features such as exons where the residues can be derived from the featureloc. It is recommended that the value for this column be manifested for features which may may non-contiguous sublocations (e.g. transcripts), since derivation at query time is non-trivial. For expressed sequence, the DNA sequence should be used rather than the RNA sequence.<br></td>
	</tr>
	<tr>
		<td style="width:150px">seqlen</td>
		<td style="width:50px">int</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>The length of the residue feature. See column:residues. This column is partially redundant with the residues column, and also with featureloc. This column is required because the location may be unknown and the residue sequence may not be manifested, yet it may be desirable to store and query the length of the feature. The seqlen should always be manifested where the length of the sequence is known.<br></td>
	</tr>
	<tr>
		<td style="width:150px">strand</td>
		<td style="width:50px">enum</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>The orientation/directionality of the location. Should be 0, -1 or +1.<br></td>
	</tr>
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column identifier should unique.</td></tr>
</table>
</p>


<h3 id="protein_entity">File: protein.txt</h3> 
<p>Contents:<br> Serves as a view on SequencesCharacteristics that are
				proteins / aa.
			
</p>
<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">identifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>user supplied or automatically assigned (using a decorator) unique and short identifier, e.g. MA1234.<br></td>
	</tr>
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>human readible name, not necessary unique.<br></td>
	</tr>
	<tr>
		<td style="width:150px">description</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>(Optional) Rudimentary meta data about the observable feature. Use of ontology       terms references to establish unambigious descriptions is recommended.<br></td>
	</tr>
	<tr>
		<td style="width:150px">cdna_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>The gene this element lies on..
		This xref uses {cdna_identifier} to find related elements in file <a href="#gene_entity">gene.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">cdna_start</td>
		<td style="width:50px">int</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Start position on cDNA sequence.<br></td>
	</tr>
	<tr>
		<td style="width:150px">cdna_end</td>
		<td style="width:50px">int</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>End position on cDNA sequence.<br></td>
	</tr>
	<tr>
		<td style="width:150px">residues</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>A sequence of alphabetic characters representing biological residues (nucleic acids, amino acids). This column does not need to be manifested for all features; it is optional for features such as exons where the residues can be derived from the featureloc. It is recommended that the value for this column be manifested for features which may may non-contiguous sublocations (e.g. transcripts), since derivation at query time is non-trivial. For expressed sequence, the DNA sequence should be used rather than the RNA sequence.<br></td>
	</tr>
	<tr>
		<td style="width:150px">seqlen</td>
		<td style="width:50px">int</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>The length of the residue feature. See column:residues. This column is partially redundant with the residues column, and also with featureloc. This column is required because the location may be unknown and the residue sequence may not be manifested, yet it may be desirable to store and query the length of the feature. The seqlen should always be manifested where the length of the sequence is known.<br></td>
	</tr>
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column identifier should unique.</td></tr>
</table>
</p>


<h3 id="proteinDomain_entity">File: proteindomain.txt</h3> 
<p>Contents:<br> Serves as a view on SequenceCharacteristics that are
				protein domains.
			
</p>
<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">identifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>user supplied or automatically assigned (using a decorator) unique and short identifier, e.g. MA1234.<br></td>
	</tr>
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>human readible name, not necessary unique.<br></td>
	</tr>
	<tr>
		<td style="width:150px">description</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>(Optional) Rudimentary meta data about the observable feature. Use of ontology       terms references to establish unambigious descriptions is recommended.<br></td>
	</tr>
	<tr>
		<td style="width:150px">cdna_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>The gene this element lies on..
		This xref uses {cdna_identifier} to find related elements in file <a href="#gene_entity">gene.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">cdna_start</td>
		<td style="width:50px">int</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Start position on cDNA sequence.<br></td>
	</tr>
	<tr>
		<td style="width:150px">cdna_end</td>
		<td style="width:50px">int</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>End position on cDNA sequence.<br></td>
	</tr>
	<tr>
		<td style="width:150px">gdna_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>The genome this element lies on..
		This xref uses {gdna_identifier} to find related elements in file <a href="#chromosome_entity">chromosome.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">gdna_start</td>
		<td style="width:50px">int</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Start position on genomic sequence.<br></td>
	</tr>
	<tr>
		<td style="width:150px">gdna_end</td>
		<td style="width:50px">int</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>End position on genomic sequence.<br></td>
	</tr>
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column identifier should unique.</td></tr>
</table>
</p>


<h3 id="exon_entity">File: exon.txt</h3> 
<p>Contents:<br> Serves as a view on SequenceCharacteristics that are
				exons or introns.
			
</p>
<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">identifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>user supplied or automatically assigned (using a decorator) unique and short identifier, e.g. MA1234.<br></td>
	</tr>
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>human readible name, not necessary unique.<br></td>
	</tr>
	<tr>
		<td style="width:150px">description</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>(Optional) Rudimentary meta data about the observable feature. Use of ontology       terms references to establish unambigious descriptions is recommended.<br></td>
	</tr>
	<tr>
		<td style="width:150px">cdna_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>The gene this element lies on..
		This xref uses {cdna_identifier} to find related elements in file <a href="#gene_entity">gene.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">cdna_start</td>
		<td style="width:50px">int</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Start position on cDNA sequence.<br></td>
	</tr>
	<tr>
		<td style="width:150px">cdna_end</td>
		<td style="width:50px">int</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>End position on cDNA sequence.<br></td>
	</tr>
	<tr>
		<td style="width:150px">gdna_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>The genome this element lies on..
		This xref uses {gdna_identifier} to find related elements in file <a href="#chromosome_entity">chromosome.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">gdna_start</td>
		<td style="width:50px">int</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Start position on genomic sequence.<br></td>
	</tr>
	<tr>
		<td style="width:150px">gdna_end</td>
		<td style="width:50px">int</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>End position on genomic sequence.<br></td>
	</tr>
	<tr>
		<td style="width:150px">isintron</td>
		<td style="width:50px">bool</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Is this exon actually an intron?.<br></td>
	</tr>
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column identifier should unique.</td></tr>
</table>
</p>


<h3 id="variant_entity">File: variant.txt</h3> 
<p>Contents:<br> Convenient entity to define in one place variants on
				gdna, cdna and aa level.
			
</p>
<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">identifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>user supplied or automatically assigned (using a decorator) unique and short identifier, e.g. MA1234.<br></td>
	</tr>
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>human readible name, not necessary unique.<br></td>
	</tr>
	<tr>
		<td style="width:150px">description</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>(Optional) Rudimentary meta data about the observable feature. Use of ontology       terms references to establish unambigious descriptions is recommended.<br></td>
	</tr>
	<tr>
		<td style="width:150px">gdna_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>The genome this element lies on..
		This xref uses {gdna_identifier} to find related elements in file <a href="#chromosome_entity">chromosome.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">gdna_start</td>
		<td style="width:50px">int</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Start position on genomic sequence.<br></td>
	</tr>
	<tr>
		<td style="width:150px">gdna_end</td>
		<td style="width:50px">int</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>End position on genomic sequence.<br></td>
	</tr>
	<tr>
		<td style="width:150px">cdna_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>The gene this element lies on..
		This xref uses {cdna_identifier} to find related elements in file <a href="#gene_entity">gene.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">cdna_start</td>
		<td style="width:50px">int</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Start position on cDNA sequence.<br></td>
	</tr>
	<tr>
		<td style="width:150px">cdna_end</td>
		<td style="width:50px">int</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>End position on cDNA sequence.<br></td>
	</tr>
	<tr>
		<td style="width:150px">aa_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>The protein sequence this element lies on..
		This xref uses {aa_identifier} to find related elements in file <a href="#protein_entity">protein.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">aa_start</td>
		<td style="width:50px">int</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Start position on amino acid sequence.<br></td>
	</tr>
	<tr>
		<td style="width:150px">aa_end</td>
		<td style="width:50px">int</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>End position on amino acid sequence.<br></td>
	</tr>
	<tr>
		<td style="width:150px">gdna_notation</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>gDNA notation of the variant, e.g. g.1234567C>T.<br></td>
	</tr>
	<tr>
		<td style="width:150px">cdna_notation</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>cDNA notation of the variant, e.g. c.123C>T.<br></td>
	</tr>
	<tr>
		<td style="width:150px">aa_notation</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Aa notation of the variant, e.g. p.Ser123ArgfsX12.<br></td>
	</tr>
	<tr>
		<td style="width:150px">varianttype_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Type of the variant..
		This xref uses {varianttype_identifier} to find related elements in file <a href="#ontologyTerm_entity">ontologyTerm.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column identifier should unique.</td></tr>
</table>
</p>


<h2>org.molgenis.organization file types</h2>
<p>Generic entities to organize data in studies, experiments
			and data sets.
		
</p>
<h3 id="study_entity">File: study.txt</h3> 
<p>Contents:<br> Investigation defines self-contained units of study. For
				example: Framingham study. Optionally a description and an accession
				to a data source can be provided. Each Investigation has a unique
				identifier.
			
</p>
<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">identifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>user supplied or automatically assigned (using a decorator) unique and short identifier, e.g. MA1234.<br></td>
	</tr>
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>human readible name, not necessary unique.<br></td>
	</tr>
	<tr>
		<td style="width:150px">description</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>(Optional)      Rudimentary meta data about the Investigation.<br></td>
	</tr>
	<tr>
		<td style="width:150px">startdate</td>
		<td style="width:50px">datetime</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">today</td>
		<td>The start point of the study.<br></td>
	</tr>
	<tr>
		<td style="width:150px">updatedate</td>
		<td style="width:50px">datetime</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">today</td>
		<td>Last time the investigation was modified.<br></td>
	</tr>
	<tr>
		<td style="width:150px">enddate</td>
		<td style="width:50px">datetime</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>The end point of the study.<br></td>
	</tr>
	<tr>
		<td style="width:150px">contact_Name<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Primary contact person for this study.
		This xref uses {contact_name} to find related elements in file <a href="#person_entity">person.txt</a> based on unique column {name}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">partofinvestigation_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>xref to the investigation the study is part of.
		This xref uses {partofinvestigation_identifier} to find related elements in file <a href="#investigation_entity">investigation.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column identifier should unique.</td></tr>
</table>
</p>


<h3 id="experiment_entity">File: experiment.txt</h3> 

<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">identifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>user supplied or automatically assigned (using a decorator) unique and short identifier, e.g. MA1234.<br></td>
	</tr>
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>human readible name, not necessary unique.<br></td>
	</tr>
	<tr>
		<td style="width:150px">study_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Part of Study..
		This xref uses {study_identifier} to find related elements in file <a href="#study_entity">study.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">design</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Design.<br></td>
	</tr>
	<tr>
		<td style="width:150px">experimenttype_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Experiment type. E.g. 'case-control'..
		This xref uses {experimenttype_identifier} to find related elements in file <a href="#ontologyTerm_entity">ontologyTerm.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">totalmarkerstested</td>
		<td style="width:50px">int</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Total markers tested.<br></td>
	</tr>
	<tr>
		<td style="width:150px">totalmarkersimported</td>
		<td style="width:50px">int</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Total markers imported.<br></td>
	</tr>
	<tr>
		<td style="width:150px">objective</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Objective.<br></td>
	</tr>
	<tr>
		<td style="width:150px">outcome</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Outcome.<br></td>
	</tr>
	<tr>
		<td style="width:150px">comments</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Comments.<br></td>
	</tr>
	<tr>
		<td style="width:150px">individualdatastatement</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">Access to individual-level data must be made to the study authors</td>
		<td>Individual data statement.<br></td>
	</tr>
	<tr>
		<td style="width:150px">timecreated</td>
		<td style="width:50px">datetime</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Time created.<br></td>
	</tr>
	<tr>
		<td style="width:150px">assayedpanels_Identifier<br></td>
		<td style="width:50px">mref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Assayed panels.
		This mref uses {assayedpanels_identifier} to find related elements in file <a href="#panel_entity">panel.txt</a> based on unique column {identifier}.
		. More than one reference can be added separated by '|', for example: ref1|ref2|ref3.</td>
	</tr>	
	<tr>
		<td style="width:150px">datasets_Identifier<br></td>
		<td style="width:50px">mref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>DataSets that were input/output of this experiment..
		This mref uses {datasets_identifier} to find related elements in file <a href="#dataSet_entity">dataSet.txt</a> based on unique column {identifier}.
		. More than one reference can be added separated by '|', for example: ref1|ref2|ref3.</td>
	</tr>	
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column identifier should unique.</td></tr>
</table>
</p>


<h3 id="institute_entity">File: institute.txt</h3> 
<p>Contents:<br>  A contact is either a person or an organization. Copied
				from FuGE::Contact.
			
</p>
<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>name.<br></td>
	</tr>
	<tr>
		<td style="width:150px">address</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>The address of the Contact.<br></td>
	</tr>
	<tr>
		<td style="width:150px">phone</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>The telephone number of the Contact including the suitable area codes.<br></td>
	</tr>
	<tr>
		<td style="width:150px">city</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Added from the old definition of MolgenisUser. City of this contact.<br></td>
	</tr>
	<tr>
		<td style="width:150px">country</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Added from the old definition of MolgenisUser. Country of this contact.<br></td>
	</tr>
	<tr>
		<td style="width:150px">fax</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>The fax number of the Contact.<br></td>
	</tr>
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column name should unique.</td></tr>
</table>
</p>


<h3 id="person_entity">File: person.txt</h3> 
<p>Contents:<br> 

				Person represents one or more people involved with an Investigation.
				This may include authors on a paper, lab personnel or PIs. Person
				has last name, firstname, mid initial, address, contact and email. A
				Person role is included to represent how a Person is involved with
				an investigation. For submission to repository purposes an allowed
				value is 'submitter' and the term is present in the MGED Ontology,
				an alternative use could represent job title. An Example from
				ArrayExpress is E-MTAB-506
				<a href="ftp://ftp.ebi.ac.uk/pub/databases/microarray/data/experiment/TABM/E-TABM-506/E-TABM-506.idf.txt">
					ftp://ftp.ebi.ac.uk/pub/databases/microarray/data/experiment/TABM/E-TABM-506/E-TABM-506.idf.txt.
				</a>
				.
    <!-- From MIBBI : Name should either be of the person or the institutional role holding responsibility for the associated (meta)data. -->
    <br/>
				The FUGE equivalent to Person is FuGE::Person.
			
</p>
<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Name.<br></td>
	</tr>
	<tr>
		<td style="width:150px">title</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>An academic title, e.g. Prof.dr, PhD.<br></td>
	</tr>
	<tr>
		<td style="width:150px">firstname</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>First Name.<br></td>
	</tr>
	<tr>
		<td style="width:150px">midinitials</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Mid Initials.<br></td>
	</tr>
	<tr>
		<td style="width:150px">lastname</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Last Name.<br></td>
	</tr>
	<tr>
		<td style="width:150px">email</td>
		<td style="width:50px">email</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Email.<br></td>
	</tr>
	<tr>
		<td style="width:150px">phone</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>The telephone number of the Contact including the suitable area codes.<br></td>
	</tr>
	<tr>
		<td style="width:150px">primaryaffilation_name<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Affliations.
		This xref uses {primaryaffilation_name} to find related elements in file <a href="#institute_entity">institute.txt</a> based on unique column {name}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">affiliateinstitutions_name<br></td>
		<td style="width:50px">mref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Affliated Institutes.
		This mref uses {affiliateinstitutions_name} to find related elements in file <a href="#institute_entity">institute.txt</a> based on unique column {name}.
		. More than one reference can be added separated by '|', for example: ref1|ref2|ref3.</td>
	</tr>	
	<tr>
		<td style="width:150px">orcidpersonreference_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>OrcidPersonReference.
		This xref uses {orcidpersonreference_identifier} to find related elements in file <a href="#ontologyTerm_entity">ontologyTerm.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column name should unique.</td></tr>
<tr><td colspan="5">Contraint: values in the combined columns (firstname, midinitials, lastname) should be unique.</td></tr>
</table>
</p>


<h3 id="citation_entity">File: citation.txt</h3> 
<p>Contents:<br> 

				Citation of a document or data set. Publications have
				attributes of
				publications Authors and also DOI and Pubmed
				identifiers (when these
				are available). These are represented as
				OntologyTerms as in the
				MAGE-TAB model all 'xrefs' (cross
				references) for ontologies and
				accession numbers are handled
				generically. An example of a
				publication is available in an IDF file
				from ArrayExpress is
				experiment E-MTAB-506
				<a href="ftp://ftp.ebi.ac.uk/pub/databases/microarray/data/experiment/TABM/E-TABM-506/E-TABM-506.idf.txt">ftp://ftp.ebi.ac.uk/pub/databases/microarray/data/experiment/TABM/E-TABM-506/E-TABM-506.idf.txt
				</a>
				.
				<br/>
				The FuGE equivalent to Publication is FuGE::Bibliographic Reference.
			
</p>
<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">identifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>user supplied or automatically assigned (using a decorator) unique and short identifier, e.g. MA1234.<br></td>
	</tr>
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>human readible name, not necessary unique.<br></td>
	</tr>
	<tr>
		<td style="width:150px">pubmedid</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Pubmed ID.<br></td>
	</tr>
	<tr>
		<td style="width:150px">doi</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Publication DOI.<br></td>
	</tr>
	<tr>
		<td style="width:150px">ontologyterms_Identifier<br></td>
		<td style="width:50px">mref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>ontology terms such as MeSH.
		This mref uses {ontologyterms_identifier} to find related elements in file <a href="#ontologyTerm_entity">ontologyTerm.txt</a> based on unique column {identifier}.
		. More than one reference can be added separated by '|', for example: ref1|ref2|ref3.</td>
	</tr>	
	<tr>
		<td style="width:150px">authorlist</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>The names of the authors of the publication.<br></td>
	</tr>
	<tr>
		<td style="width:150px">title</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>The title of the Publication.<br></td>
	</tr>
	<tr>
		<td style="width:150px">description</td>
		<td style="width:50px">text</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Description.<br></td>
	</tr>
	<tr>
		<td style="width:150px">status_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>The status of the Publication.
		This xref uses {status_identifier} to find related elements in file <a href="#ontologyTerm_entity">ontologyTerm.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column identifier should unique.</td></tr>
<tr><td colspan="5">Constraint: values in column pubmedid should unique.</td></tr>
<tr><td colspan="5">Constraint: values in column doi should unique.</td></tr>
</table>
</p>


<h3 id="contribution_entity">File: contribution.txt</h3> 
<p>Contents:<br> A contribution describes the part a Person has
				contributed to a Contributable item.
			
</p>
<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">identifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>user supplied or automatically assigned (using a decorator) unique and short identifier, e.g. MA1234.<br></td>
	</tr>
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>human readible name, not necessary unique.<br></td>
	</tr>
	<tr>
		<td style="width:150px">researcher_Name<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>The person involved.
		This xref uses {researcher_name} to find related elements in file <a href="#person_entity">person.txt</a> based on unique column {name}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">submission_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>The submission contributed to..
		This xref uses {submission_identifier} to find related elements in file <a href="#submission_entity">submission.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">issubmitter</td>
		<td style="width:50px">enum</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Submitter?.<br></td>
	</tr>
	<tr>
		<td style="width:150px">isauthor</td>
		<td style="width:50px">enum</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Author?.<br></td>
	</tr>
	<tr>
		<td style="width:150px">issource</td>
		<td style="width:50px">enum</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Source?.<br></td>
	</tr>
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column identifier should unique.</td></tr>
</table>
</p>


<h3 id="submission_entity">File: submission.txt</h3> 

<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">identifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>user supplied or automatically assigned (using a decorator) unique and short identifier, e.g. MA1234.<br></td>
	</tr>
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>human readible name, not necessary unique.<br></td>
	</tr>
	<tr>
		<td style="width:150px">timecreated</td>
		<td style="width:50px">datetime</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Time created.<br></td>
	</tr>
	<tr>
		<td style="width:150px">study_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Study.
		This xref uses {study_identifier} to find related elements in file <a href="#study_entity">study.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column identifier should unique.</td></tr>
</table>
</p>


<h2>org.molgenis.gwascentral file types</h2>

<h3 id="investigation_entity">File: investigation.txt</h3> 

<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">identifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>user supplied or automatically assigned (using a decorator) unique and short identifier, e.g. MA1234.<br></td>
	</tr>
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>human readible name, not necessary unique.<br></td>
	</tr>
	<tr>
		<td style="width:150px">title</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Nice title of Investigation.<br></td>
	</tr>
	<tr>
		<td style="width:150px">shortname</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Shorthand name for layout.<br></td>
	</tr>
	<tr>
		<td style="width:150px">version</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Accession version.<br></td>
	</tr>
	<tr>
		<td style="width:150px">background</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Short piece of information describing why the study is taking place, e.g. risk factors for a population.<br></td>
	</tr>
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column identifier should unique.</td></tr>
</table>
</p>


<h3 id="studyDetails_entity">File: studydetails.txt</h3> 

<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">study_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Study.
		This xref uses {study_identifier} to find related elements in file <a href="#study_entity">study.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">title</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Nice title of the paper.<br></td>
	</tr>
	<tr>
		<td style="width:150px">shortname</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Shorthand name for layout.<br></td>
	</tr>
	<tr>
		<td style="width:150px">studyabstract</td>
		<td style="width:50px">text</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Abstract.<br></td>
	</tr>
	<tr>
		<td style="width:150px">version</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Accession version.<br></td>
	</tr>
	<tr>
		<td style="width:150px">background</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Short piece of information describing why the study is taking place, e.g. risk factors for a population.<br></td>
	</tr>
	<tr>
		<td style="width:150px">objectives</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>What this study aims to achieve.<br></td>
	</tr>
	<tr>
		<td style="width:150px">keyresults</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Noticable results from this study.<br></td>
	</tr>
	<tr>
		<td style="width:150px">conclusions</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Description of the conclusions drawn.<br></td>
	</tr>
	<tr>
		<td style="width:150px">studydesign</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Study design.<br></td>
	</tr>
	<tr>
		<td style="width:150px">studysizereason</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Reason for study size.<br></td>
	</tr>
	<tr>
		<td style="width:150px">studypower</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Study power.<br></td>
	</tr>
	<tr>
		<td style="width:150px">sourcesofbias</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Sources of bias.<br></td>
	</tr>
	<tr>
		<td style="width:150px">limitations</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Limitations.<br></td>
	</tr>
	<tr>
		<td style="width:150px">acknowledgements</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Acknowledgements.<br></td>
	</tr>
	<tr>
		<td style="width:150px">primarycitation_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>primaryCitation.
		This xref uses {primarycitation_identifier} to find related elements in file <a href="#citation_entity">citation.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">othercitations_Identifier<br></td>
		<td style="width:50px">mref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Contact persons for this study.
		This mref uses {othercitations_identifier} to find related elements in file <a href="#citation_entity">citation.txt</a> based on unique column {identifier}.
		. More than one reference can be added separated by '|', for example: ref1|ref2|ref3.</td>
	</tr>	
	<tr>
		<td style="width:150px">accession</td>
		<td style="width:50px">hyperlink</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>(Optional) URI or accession number to indicate source of Study. E.g. arrayexpress:M-EXP-2345.<br></td>
	</tr>
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column study should unique.</td></tr>
</table>
</p>


<h3 id="frequencyCluster_entity">File: frequencycluster.txt</h3> 

<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">identifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>user supplied or automatically assigned (using a decorator) unique and short identifier, e.g. MA1234.<br></td>
	</tr>
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>human readible name, not necessary unique.<br></td>
	</tr>
	<tr>
		<td style="width:150px">description</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>(Optional) Rudimentary meta data about the observable feature. Use of ontology       terms references to establish unambigious descriptions is recommended.<br></td>
	</tr>
	<tr>
		<td style="width:150px">protocolused_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Reference to the protocol that is being used (if available).
		This xref uses {protocolused_identifier} to find related elements in file <a href="#protocol_entity">protocol.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">starttime</td>
		<td style="width:50px">datetime</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">today</td>
		<td>time when the protocol started.<br></td>
	</tr>
	<tr>
		<td style="width:150px">endtime</td>
		<td style="width:50px">datetime</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">today</td>
		<td>(Optional) time when the protocol ended.<br></td>
	</tr>
	<tr>
		<td style="width:150px">dataset_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Result set identifier.
		This xref uses {dataset_identifier} to find related elements in file <a href="#dataSet_entity">dataSet.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">usedmarkerset_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Used marker set ID.
		This xref uses {usedmarkerset_identifier} to find related elements in file <a href="#usedMarkerSet_entity">usedMarkerSet.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">markerid</td>
		<td style="width:50px">int</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Marker ID.<br></td>
	</tr>
	<tr>
		<td style="width:150px">numberofgenotypedsamples</td>
		<td style="width:50px">int</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Number of genotyped samples.<br></td>
	</tr>
	<tr>
		<td style="width:150px">pvaluehwe</td>
		<td style="width:50px">decimal</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>P-value HWE.<br></td>
	</tr>
	<tr>
		<td style="width:150px">unadjustedpvalue</td>
		<td style="width:50px">decimal</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Unadjusted p-value.<br></td>
	</tr>
	<tr>
		<td style="width:150px">oddsratiostatement</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Odds ratio statement.<br></td>
	</tr>
	<tr>
		<td style="width:150px">attributableriskstatement</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Attributable risk statement.<br></td>
	</tr>
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column identifier should unique.</td></tr>
</table>
</p>


<h3 id="genotypeFrequency_entity">File: genotypefrequency.txt</h3> 

<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">identifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>user supplied or automatically assigned (using a decorator) unique and short identifier, e.g. MA1234.<br></td>
	</tr>
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>human readible name, not necessary unique.<br></td>
	</tr>
	<tr>
		<td style="width:150px">description</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>(Optional) Rudimentary meta data about the observable feature. Use of ontology       terms references to establish unambigious descriptions is recommended.<br></td>
	</tr>
	<tr>
		<td style="width:150px">protocolused_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Reference to the protocol that is being used (if available).
		This xref uses {protocolused_identifier} to find related elements in file <a href="#protocol_entity">protocol.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">starttime</td>
		<td style="width:50px">datetime</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">today</td>
		<td>time when the protocol started.<br></td>
	</tr>
	<tr>
		<td style="width:150px">endtime</td>
		<td style="width:50px">datetime</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">today</td>
		<td>(Optional) time when the protocol ended.<br></td>
	</tr>
	<tr>
		<td style="width:150px">frequencycluster_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Frequency cluster ID.
		This xref uses {frequencycluster_identifier} to find related elements in file <a href="#frequencyCluster_entity">frequencyCluster.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">genotypecombo</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Genotype combo.<br></td>
	</tr>
	<tr>
		<td style="width:150px">frequencyasproportion</td>
		<td style="width:50px">decimal</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Frequency as proportion.<br></td>
	</tr>
	<tr>
		<td style="width:150px">numbersampleswithgenotype</td>
		<td style="width:50px">int</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Number of samples with genotype.<br></td>
	</tr>
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column identifier should unique.</td></tr>
</table>
</p>


<h3 id="alleleFrequency_entity">File: allelefrequency.txt</h3> 

<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">identifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>user supplied or automatically assigned (using a decorator) unique and short identifier, e.g. MA1234.<br></td>
	</tr>
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>human readible name, not necessary unique.<br></td>
	</tr>
	<tr>
		<td style="width:150px">description</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>(Optional) Rudimentary meta data about the observable feature. Use of ontology       terms references to establish unambigious descriptions is recommended.<br></td>
	</tr>
	<tr>
		<td style="width:150px">protocolused_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Reference to the protocol that is being used (if available).
		This xref uses {protocolused_identifier} to find related elements in file <a href="#protocol_entity">protocol.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">starttime</td>
		<td style="width:50px">datetime</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">today</td>
		<td>time when the protocol started.<br></td>
	</tr>
	<tr>
		<td style="width:150px">endtime</td>
		<td style="width:50px">datetime</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">today</td>
		<td>(Optional) time when the protocol ended.<br></td>
	</tr>
	<tr>
		<td style="width:150px">frequencycluster_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Frequency cluster ID.
		This xref uses {frequencycluster_identifier} to find related elements in file <a href="#frequencyCluster_entity">frequencyCluster.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">allelecombo</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Allele combo.<br></td>
	</tr>
	<tr>
		<td style="width:150px">frequencyasproportion</td>
		<td style="width:50px">decimal</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Frequency as proportion.<br></td>
	</tr>
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column identifier should unique.</td></tr>
</table>
</p>


<h3 id="phenotypeProperty_entity">File: phenotypeproperty.txt</h3> 

<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">identifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>user supplied or automatically assigned (using a decorator) unique and short identifier, e.g. MA1234.<br></td>
	</tr>
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Phenotype Property.<br></td>
	</tr>
	<tr>
		<td style="width:150px">description</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>(Optional) Rudimentary meta data about the observable feature. Use of ontology       terms references to establish unambigious descriptions is recommended.<br></td>
	</tr>
	<tr>
		<td style="width:150px">unit_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>(Optional) Reference to the well-defined measurement unit used to observe this feature       (if feature is that concrete). E.g. mmHg.
		This xref uses {unit_identifier} to find related elements in file <a href="#ontologyTerm_entity">ontologyTerm.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">datatype</td>
		<td style="width:50px">enum</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">string</td>
		<td>(Optional) Reference to the technical data type. E.g. 'int'.<br></td>
	</tr>
	<tr>
		<td style="width:150px">temporal</td>
		<td style="width:50px">bool</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">false</td>
		<td>Whether this feature is time dependent and can have different values when measured       on different times (e.g. weight, temporal=true) or generally only measured once (e.g. birth date,       temporal=false).<br></td>
	</tr>
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column identifier should unique.</td></tr>
</table>
</p>


<h3 id="phenotypeMethod_entity">File: phenotypemethod.txt</h3> 

<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">identifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>user supplied or automatically assigned (using a decorator) unique and short identifier, e.g. MA1234.<br></td>
	</tr>
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Name.<br></td>
	</tr>
	<tr>
		<td style="width:150px">description</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>(Optional) Rudimentary meta data about the observable feature. Use of ontology       terms references to establish unambigious descriptions is recommended.<br></td>
	</tr>
	<tr>
		<td style="width:150px">protocolused_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Reference to the protocol that is being used (if available).
		This xref uses {protocolused_identifier} to find related elements in file <a href="#protocol_entity">protocol.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">starttime</td>
		<td style="width:50px">datetime</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">today</td>
		<td>time when the protocol started.<br></td>
	</tr>
	<tr>
		<td style="width:150px">endtime</td>
		<td style="width:50px">datetime</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">today</td>
		<td>(Optional) time when the protocol ended.<br></td>
	</tr>
	<tr>
		<td style="width:150px">studyid_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Points to study that this method came in.
		This xref uses {studyid_identifier} to find related elements in file <a href="#study_entity">study.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">phenotypepropertyid_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Phenotype property name.
		This xref uses {phenotypepropertyid_identifier} to find related elements in file <a href="#phenotypeProperty_entity">phenotypeProperty.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">sample</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Biological system or sample type being assessed.<br></td>
	</tr>
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column identifier should unique.</td></tr>
</table>
</p>


<h3 id="phenotypeValue_entity">File: phenotypevalue.txt</h3> 

<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">observationset_id<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Reference to the observation. For example a particular patient visit or the application of a microarray or the calculation of a QTL model.
		This xref uses {observationset_id} to find related elements in file <a href="#observationSet_entity">observationSet.txt</a> based on unique column {id}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">feature_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>References the ObservableFeature that this observation was made on. For example 'probe123'..
		This xref uses {feature_identifier} to find related elements in file <a href="#observableFeature_entity">observableFeature.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">characteristic_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Is brother of [characteristic], or Average of [height].
		This xref uses {characteristic_identifier} to find related elements in file <a href="#characteristic_entity">characteristic.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">value</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Value.<br></td>
	</tr>
	<tr>
		<td style="width:150px">identifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>user supplied or automatically assigned (using a decorator) unique and short identifier, e.g. MA1234.<br></td>
	</tr>
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Name.<br></td>
	</tr>
	<tr>
		<td style="width:150px">phenotypepropertyid_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Phenotype property name.
		This xref uses {phenotypepropertyid_identifier} to find related elements in file <a href="#phenotypeProperty_entity">phenotypeProperty.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">valuerank</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Value Rank.<br></td>
	</tr>
	<tr>
		<td style="width:150px">valueismean</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Value is mean.<br></td>
	</tr>
	<tr>
		<td style="width:150px">std</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Standard Deviation.<br></td>
	</tr>
	<tr>
		<td style="width:150px">min</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Minimum value.<br></td>
	</tr>
	<tr>
		<td style="width:150px">max</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Maximum value.<br></td>
	</tr>
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column identifier should unique.</td></tr>
</table>
</p>


<h3 id="samplePanel_entity">File: samplepanel.txt</h3> 

<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">identifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>user supplied or automatically assigned (using a decorator) unique and short identifier, e.g. MA1234.<br></td>
	</tr>
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Name.<br></td>
	</tr>
	<tr>
		<td style="width:150px">description</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Description.<br></td>
	</tr>
	<tr>
		<td style="width:150px">paneltype_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Indicate the type of Panel (example: Sample panel, AssayedPanel, Natural=wild type, Parental=parents of a cross, F1=First generation of cross, RCC=Recombinant congenic, CSS=chromosome substitution).
		This xref uses {paneltype_identifier} to find related elements in file <a href="#ontologyTerm_entity">ontologyTerm.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">numberofindividuals</td>
		<td style="width:50px">int</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>NumberOfIndividuals.<br></td>
	</tr>
	<tr>
		<td style="width:150px">species_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>The species this panel is an instance of/part of/extracted from..
		This xref uses {species_identifier} to find related elements in file <a href="#species_entity">species.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">individuals_Identifier<br></td>
		<td style="width:50px">mref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>The list of individuals in this panel.
		This mref uses {individuals_identifier} to find related elements in file <a href="#individual_entity">individual.txt</a> based on unique column {identifier}.
		. More than one reference can be added separated by '|', for example: ref1|ref2|ref3.</td>
	</tr>	
	<tr>
		<td style="width:150px">centralidentifier_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>The central GWAS identifier for this panel.
		This xref uses {centralidentifier_identifier} to find related elements in file <a href="#ontologyTerm_entity">ontologyTerm.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">composition</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Composition.<br></td>
	</tr>
	<tr>
		<td style="width:150px">totalnumberofindividuals</td>
		<td style="width:50px">int</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>*...but required for association datasets.<br></td>
	</tr>
	<tr>
		<td style="width:150px">numberofsexmale</td>
		<td style="width:50px">int</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Number of males.<br></td>
	</tr>
	<tr>
		<td style="width:150px">numberofsexfemale</td>
		<td style="width:50px">int</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Number of females.<br></td>
	</tr>
	<tr>
		<td style="width:150px">numberofsexunknown</td>
		<td style="width:50px">int</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Number of unknown sex.<br></td>
	</tr>
	<tr>
		<td style="width:150px">numberofprobands</td>
		<td style="width:50px">int</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>field used only if Composition = 'Trios'.<br></td>
	</tr>
	<tr>
		<td style="width:150px">numberofparents</td>
		<td style="width:50px">int</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>field used only if Composition = 'Trios'.<br></td>
	</tr>
	<tr>
		<td style="width:150px">modeofrecruitment</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Mode of recruitment.<br></td>
	</tr>
	<tr>
		<td style="width:150px">diagnosisagerange</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Diagnosis age range.<br></td>
	</tr>
	<tr>
		<td style="width:150px">diagnosisperiod</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Diagnosis period.<br></td>
	</tr>
	<tr>
		<td style="width:150px">samplingagerange</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Sampling age range.<br></td>
	</tr>
	<tr>
		<td style="width:150px">samplingperiod</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Sampling period.<br></td>
	</tr>
	<tr>
		<td style="width:150px">populationinfo</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Population information.<br></td>
	</tr>
	<tr>
		<td style="width:150px">geographicregioninfo</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Geographic region information.<br></td>
	</tr>
	<tr>
		<td style="width:150px">ethnicityinfo</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Ethnicity information.<br></td>
	</tr>
	<tr>
		<td style="width:150px">birthplaceinfo</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Birth place information.<br></td>
	</tr>
	<tr>
		<td style="width:150px">admixtureinfo</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Admixture information.<br></td>
	</tr>
	<tr>
		<td style="width:150px">environmentinfo</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Environment information.<br></td>
	</tr>
	<tr>
		<td style="width:150px">sourceofdna</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>In (SELECT [SourceOfDNA] FROM [Sampleset_SourceOfDNAList];).<br></td>
	</tr>
	<tr>
		<td style="width:150px">dnasarepooled</td>
		<td style="width:50px">enum</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">Undefined</td>
		<td>Are DNAs pooled?.<br></td>
	</tr>
	<tr>
		<td style="width:150px">dnasarewga</td>
		<td style="width:50px">enum</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">Undefined</td>
		<td>Are DNAs WGA?.<br></td>
	</tr>
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column identifier should unique.</td></tr>
</table>
</p>


<h3 id="assayedPanel_entity">File: assayedpanel.txt</h3> 

<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">identifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>user supplied or automatically assigned (using a decorator) unique and short identifier, e.g. MA1234.<br></td>
	</tr>
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Name.<br></td>
	</tr>
	<tr>
		<td style="width:150px">description</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Description.<br></td>
	</tr>
	<tr>
		<td style="width:150px">paneltype_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Indicate the type of Panel (example: Sample panel, AssayedPanel, Natural=wild type, Parental=parents of a cross, F1=First generation of cross, RCC=Recombinant congenic, CSS=chromosome substitution).
		This xref uses {paneltype_identifier} to find related elements in file <a href="#ontologyTerm_entity">ontologyTerm.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">numberofindividuals</td>
		<td style="width:50px">int</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>NumberOfIndividuals.<br></td>
	</tr>
	<tr>
		<td style="width:150px">species_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>The species this panel is an instance of/part of/extracted from..
		This xref uses {species_identifier} to find related elements in file <a href="#species_entity">species.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">individuals_Identifier<br></td>
		<td style="width:50px">mref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>The list of individuals in this panel.
		This mref uses {individuals_identifier} to find related elements in file <a href="#individual_entity">individual.txt</a> based on unique column {identifier}.
		. More than one reference can be added separated by '|', for example: ref1|ref2|ref3.</td>
	</tr>	
	<tr>
		<td style="width:150px">totalnumberofindividuals</td>
		<td style="width:50px">int</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>*...but required for association datasets.<br></td>
	</tr>
	<tr>
		<td style="width:150px">numberofsexmale</td>
		<td style="width:50px">int</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Number of males.<br></td>
	</tr>
	<tr>
		<td style="width:150px">numberofsexfemale</td>
		<td style="width:50px">int</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Number of females.<br></td>
	</tr>
	<tr>
		<td style="width:150px">numberofsexunknown</td>
		<td style="width:50px">int</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Number of unknown sex.<br></td>
	</tr>
	<tr>
		<td style="width:150px">numberofprobands</td>
		<td style="width:50px">int</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>field used only if Composition = 'Trios'.<br></td>
	</tr>
	<tr>
		<td style="width:150px">numberofparents</td>
		<td style="width:50px">int</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>field used only if Composition = 'Trios'.<br></td>
	</tr>
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column identifier should unique.</td></tr>
</table>
</p>


<h3 id="gWASExperiment_entity">File: gwasexperiment.txt</h3> 

<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">identifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>user supplied or automatically assigned (using a decorator) unique and short identifier, e.g. MA1234.<br></td>
	</tr>
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>human readible name, not necessary unique.<br></td>
	</tr>
	<tr>
		<td style="width:150px">study_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Part of Study..
		This xref uses {study_identifier} to find related elements in file <a href="#study_entity">study.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">design</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Design.<br></td>
	</tr>
	<tr>
		<td style="width:150px">experimenttype_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Experiment type. E.g. 'case-control'..
		This xref uses {experimenttype_identifier} to find related elements in file <a href="#ontologyTerm_entity">ontologyTerm.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">totalmarkerstested</td>
		<td style="width:50px">int</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Total markers tested.<br></td>
	</tr>
	<tr>
		<td style="width:150px">totalmarkersimported</td>
		<td style="width:50px">int</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Total markers imported.<br></td>
	</tr>
	<tr>
		<td style="width:150px">objective</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Objective.<br></td>
	</tr>
	<tr>
		<td style="width:150px">outcome</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Outcome.<br></td>
	</tr>
	<tr>
		<td style="width:150px">comments</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Comments.<br></td>
	</tr>
	<tr>
		<td style="width:150px">individualdatastatement</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">Access to individual-level data must be made to the study authors</td>
		<td>Individual data statement.<br></td>
	</tr>
	<tr>
		<td style="width:150px">timecreated</td>
		<td style="width:50px">datetime</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Time created.<br></td>
	</tr>
	<tr>
		<td style="width:150px">assayedpanels_Identifier<br></td>
		<td style="width:50px">mref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Assayed panels.
		This mref uses {assayedpanels_identifier} to find related elements in file <a href="#panel_entity">panel.txt</a> based on unique column {identifier}.
		. More than one reference can be added separated by '|', for example: ref1|ref2|ref3.</td>
	</tr>	
	<tr>
		<td style="width:150px">datasets_Identifier<br></td>
		<td style="width:50px">mref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>DataSets that were input/output of this experiment..
		This mref uses {datasets_identifier} to find related elements in file <a href="#dataSet_entity">dataSet.txt</a> based on unique column {identifier}.
		. More than one reference can be added separated by '|', for example: ref1|ref2|ref3.</td>
	</tr>	
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column identifier should unique.</td></tr>
</table>
</p>


<h3 id="usedMarkerSet_entity">File: usedmarkerset.txt</h3> 

<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">identifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>user supplied or automatically assigned (using a decorator) unique and short identifier, e.g. MA1234.<br></td>
	</tr>
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>human readible name, not necessary unique.<br></td>
	</tr>
	<tr>
		<td style="width:150px">description</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>(Optional) Rudimentary meta data about the observable feature. Use of ontology       terms references to establish unambigious descriptions is recommended.<br></td>
	</tr>
	<tr>
		<td style="width:150px">unit_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>(Optional) Reference to the well-defined measurement unit used to observe this feature       (if feature is that concrete). E.g. mmHg.
		This xref uses {unit_identifier} to find related elements in file <a href="#ontologyTerm_entity">ontologyTerm.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">datatype</td>
		<td style="width:50px">enum</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">string</td>
		<td>(Optional) Reference to the technical data type. E.g. 'int'.<br></td>
	</tr>
	<tr>
		<td style="width:150px">temporal</td>
		<td style="width:50px">bool</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">false</td>
		<td>Whether this feature is time dependent and can have different values when measured       on different times (e.g. weight, temporal=true) or generally only measured once (e.g. birth date,       temporal=false).<br></td>
	</tr>
	<tr>
		<td style="width:150px">experimentid_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Experiment ID.
		This xref uses {experimentid_identifier} to find related elements in file <a href="#experiment_entity">experiment.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">markeridentifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Marker identifier.<br></td>
	</tr>
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column identifier should unique.</td></tr>
<tr><td colspan="5">Contraint: values in the combined columns (experimentid, markeridentifier) should be unique.</td></tr>
</table>
</p>


<h3 id="significance_entity">File: significance.txt</h3> 

<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">identifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>user supplied or automatically assigned (using a decorator) unique and short identifier, e.g. MA1234.<br></td>
	</tr>
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>human readible name, not necessary unique.<br></td>
	</tr>
	<tr>
		<td style="width:150px">description</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>(Optional) Rudimentary meta data about the observable feature. Use of ontology       terms references to establish unambigious descriptions is recommended.<br></td>
	</tr>
	<tr>
		<td style="width:150px">protocolused_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Reference to the protocol that is being used (if available).
		This xref uses {protocolused_identifier} to find related elements in file <a href="#protocol_entity">protocol.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">starttime</td>
		<td style="width:50px">datetime</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">today</td>
		<td>time when the protocol started.<br></td>
	</tr>
	<tr>
		<td style="width:150px">endtime</td>
		<td style="width:50px">datetime</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">today</td>
		<td>(Optional) time when the protocol ended.<br></td>
	</tr>
	<tr>
		<td style="width:150px">usedmarkersetid_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Used marker set ID.
		This xref uses {usedmarkersetid_identifier} to find related elements in file <a href="#usedMarkerSet_entity">usedMarkerSet.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">neglogpvalue</td>
		<td style="width:50px">decimal</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Negative log p-value.<br></td>
	</tr>
	<tr>
		<td style="width:150px">unadjustedpvalue</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Unadjusted p-value.<br></td>
	</tr>
	<tr>
		<td style="width:150px">adjustedpvalue</td>
		<td style="width:50px">decimal</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Adjusted p-value.<br></td>
	</tr>
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column identifier should unique.</td></tr>
</table>
</p>


<h3 id="effectSize_entity">File: effectsize.txt</h3> 

<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">identifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>user supplied or automatically assigned (using a decorator) unique and short identifier, e.g. MA1234.<br></td>
	</tr>
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>human readible name, not necessary unique.<br></td>
	</tr>
	<tr>
		<td style="width:150px">description</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>(Optional) Rudimentary meta data about the observable feature. Use of ontology       terms references to establish unambigious descriptions is recommended.<br></td>
	</tr>
	<tr>
		<td style="width:150px">protocolused_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Reference to the protocol that is being used (if available).
		This xref uses {protocolused_identifier} to find related elements in file <a href="#protocol_entity">protocol.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">starttime</td>
		<td style="width:50px">datetime</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">today</td>
		<td>time when the protocol started.<br></td>
	</tr>
	<tr>
		<td style="width:150px">endtime</td>
		<td style="width:50px">datetime</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">today</td>
		<td>(Optional) time when the protocol ended.<br></td>
	</tr>
	<tr>
		<td style="width:150px">usedmarkersetid_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Used Marker ID.
		This xref uses {usedmarkersetid_identifier} to find related elements in file <a href="#usedMarkerSet_entity">usedMarkerSet.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">lower95bound</td>
		<td style="width:50px">decimal</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Lower95Bound.<br></td>
	</tr>
	<tr>
		<td style="width:150px">upper95bound</td>
		<td style="width:50px">decimal</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Upper95Bound.<br></td>
	</tr>
	<tr>
		<td style="width:150px">stderror</td>
		<td style="width:50px">decimal</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>StdError.<br></td>
	</tr>
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column identifier should unique.</td></tr>
</table>
</p>


<h3 id="selectionCriteria_entity">File: selectioncriteria.txt</h3> 
<p>Contents:<br> Create relationships between panels, founder panels,
				such as overlap,
				selection criteria, getting assayed panel from a
				sample panel, etc.
			
</p>
<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">sourcepanel_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>SourcePanel.
		This xref uses {sourcepanel_identifier} to find related elements in file <a href="#panel_entity">panel.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">targetpanel_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>TargetPanel.
		This xref uses {targetpanel_identifier} to find related elements in file <a href="#panel_entity">panel.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">numberofindividuals</td>
		<td style="width:50px">int</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>NumberOfIndividuals.<br></td>
	</tr>
	<tr>
		<td style="width:150px">details</td>
		<td style="width:50px">text</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Details.<br></td>
	</tr>
	<!-- all optional fields -->
	<!-- all the uniques -->
</table>
</p>



<!-- entities outside modules -->
<h2>org.molgenis file types</h2>
<h3 id="molgenisEntity_entity">File: molgenisentity.txt</h3> 
<p>Contents:<br> Referenceable catalog of entity names, menus, forms and
				plugins.
			
</p>
<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Name of the entity.<br></td>
	</tr>
	<tr>
		<td style="width:150px">type_</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Type of the entity.<br></td>
	</tr>
	<tr>
		<td style="width:150px">classname</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Full name of the entity.<br></td>
	</tr>
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column classname should unique.</td></tr>
<tr><td colspan="5">Contraint: values in the combined columns (name, type_) should be unique.</td></tr>
</table>
</p>


<h3 id="molgenisFile_entity">File: molgenisfile.txt</h3> 
<p>Contents:<br> 
				Helper entity to deal with files. Has a decorator to regulate
				storage and coupling to an Entity. Do not make abstract because of
				subtyping. This means the names of the subclasses will be used to
				distinguish MolgenisFiles and place them in the correct folders.
			
</p>
<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">identifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>user supplied or automatically assigned (using a decorator) unique and short identifier, e.g. MA1234.<br></td>
	</tr>
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>human readible name, not necessary unique.<br></td>
	</tr>
	<tr>
		<td style="width:150px">extension</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>The file extension. This will be mapped to MIME type at runtime. For example, a type 'png' will be served out as 'image/png'.<br></td>
	</tr>
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column identifier should unique.</td></tr>
<tr><td colspan="5">Constraint: values in column name should unique.</td></tr>
</table>
</p>


<h3 id="runtimeProperty_entity">File: runtimeproperty.txt</h3> 

<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">identifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>user supplied or automatically assigned (using a decorator) unique and short identifier, e.g. MA1234.<br></td>
	</tr>
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>human readible name, not necessary unique.<br></td>
	</tr>
	<tr>
		<td style="width:150px">value</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Value.<br></td>
	</tr>
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column identifier should unique.</td></tr>
<tr><td colspan="5">Constraint: values in column name should unique.</td></tr>
</table>
</p>


<h3 id="molgenisRole_entity">File: molgenisrole.txt</h3> 

<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>name.<br></td>
	</tr>
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column name should unique.</td></tr>
</table>
</p>


<h3 id="molgenisGroup_entity">File: molgenisgroup.txt</h3> 

<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>name.<br></td>
	</tr>
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column name should unique.</td></tr>
</table>
</p>


<h3 id="molgenisUser_entity">File: molgenisuser.txt</h3> 
<p>Contents:<br> Anyone who can login
.</p>
<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">username</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>username.<br></td>
	</tr>
	<tr>
		<td style="width:150px">password_</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">secret</td>
		<td>big fixme: password type.<br></td>
	</tr>
	<tr>
		<td style="width:150px">activationcode</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Used as alternative authentication mechanism to verify user email and/or if user has lost password.<br></td>
	</tr>
	<tr>
		<td style="width:150px">active</td>
		<td style="width:50px">bool</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">false</td>
		<td>Boolean to indicate if this account can be used to login.<br></td>
	</tr>
	<tr>
		<td style="width:150px">superuser</td>
		<td style="width:50px">bool</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">false</td>
		<td>superuser.<br></td>
	</tr>
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column username should unique.</td></tr>
</table>
</p>


<h3 id="molgenisRoleGroupLink_entity">File: molgenisrolegrouplink.txt</h3> 

<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">identifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>user supplied or automatically assigned (using a decorator) unique and short identifier, e.g. MA1234.<br></td>
	</tr>
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>human readible name, not necessary unique.<br></td>
	</tr>
	<tr>
		<td style="width:150px">group__name<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>group_.
		This xref uses {group__name} to find related elements in file <a href="#molgenisGroup_entity">molgenisGroup.txt</a> based on unique column {name}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">role__name<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>role_.
		This xref uses {role__name} to find related elements in file <a href="#molgenisRole_entity">molgenisRole.txt</a> based on unique column {name}.
		</td>
	</tr>	
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column identifier should unique.</td></tr>
</table>
</p>


<h3 id="molgenisPermission_entity">File: molgenispermission.txt</h3> 

<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">role__name<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>role_.
		This xref uses {role__name} to find related elements in file <a href="#molgenisRole_entity">molgenisRole.txt</a> based on unique column {name}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">entity_className<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>entity.
		This xref uses {entity_classname} to find related elements in file <a href="#molgenisEntity_entity">molgenisEntity.txt</a> based on unique column {classname}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">permission</td>
		<td style="width:50px">enum</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>permission.<br></td>
	</tr>
	<!-- all optional fields -->
	<!-- all the uniques -->
</table>
</p>


<h3 id="characteristic_entity">File: characteristic.txt</h3> 
<p>Contents:<br>  Characteristics are yes-no
				statements about things in
				the world. These can be used as part of an observation, as
				parameter
				of ObservableFeature ('measuredCharacteristic'). For
				example: 'What
				is allele of [Marker]', here the [Marker] is a
				characteristic. Also,
				Characteristics can be used as target of
				observation. Typical
				examples are 'Individual' or 'Panel'. But also
				'Marker' can be an
				Target when asked the question 'QTL p-value for
				[phenotype]': here
				both target and feature are characteristic, for
				example 'leave count'
				(phenotype characteristic) and 'PVV4' (marker
				characteristic).
			
</p>
<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">identifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>user supplied or automatically assigned (using a decorator) unique and short identifier, e.g. MA1234.<br></td>
	</tr>
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>human readible name, not necessary unique.<br></td>
	</tr>
	<tr>
		<td style="width:150px">description</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>(Optional) Rudimentary meta data about the observable feature. Use of ontology       terms references to establish unambigious descriptions is recommended.<br></td>
	</tr>
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column identifier should unique.</td></tr>
</table>
</p>


<h3 id="observationTarget_entity">File: observationtarget.txt</h3> 
<p>Contents:<br>  ObservationTarget defines subjects of
				observation, such
				as Individual, Panel, Sample, etc. For instance: 'target 1' IS A
				'Individual'.
			
</p>
<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">identifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>user supplied or automatically assigned (using a decorator) unique and short identifier, e.g. MA1234.<br></td>
	</tr>
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>human readible name, not necessary unique.<br></td>
	</tr>
	<tr>
		<td style="width:150px">description</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>(Optional) Rudimentary meta data about the observable feature. Use of ontology       terms references to establish unambigious descriptions is recommended.<br></td>
	</tr>
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column identifier should unique.</td></tr>
</table>
</p>


<h3 id="observableFeature_entity">File: observablefeature.txt</h3> 
<p>Contents:<br> 
				ObservableFeature defines anything that can be observed.
				<p/>
				In other words, ObservableFeature are the
				questions asked, e.g. 'What
				is Height?', 'What is Systolic blood
				pressure?', or 'Has blue eyes?'.
				<p/>
				Some questions may
				be repeated for multiple characteristics. For
				example 'What is [MarkerAllele]
				observed?' can be applied to all
				elements of a MarkerSet, and 'What
				is [medicin codes] uses' can be
				applied to a set of Medicine codes. This can be specified using the
				measuredCharacteristic field.
				<p/>
				The identifier of ObservableFeature is globally
				unique. It is
				recommended that
				each
				ObservableFeature is named
				according to a
				well-defined ontology
				term or database accession.
			
</p>
<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">identifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>user supplied or automatically assigned (using a decorator) unique and short identifier, e.g. MA1234.<br></td>
	</tr>
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>human readible name, not necessary unique.<br></td>
	</tr>
	<tr>
		<td style="width:150px">description</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>(Optional) Rudimentary meta data about the observable feature. Use of ontology       terms references to establish unambigious descriptions is recommended.<br></td>
	</tr>
	<tr>
		<td style="width:150px">unit_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>(Optional) Reference to the well-defined measurement unit used to observe this feature       (if feature is that concrete). E.g. mmHg.
		This xref uses {unit_identifier} to find related elements in file <a href="#ontologyTerm_entity">ontologyTerm.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">datatype</td>
		<td style="width:50px">enum</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">string</td>
		<td>(Optional) Reference to the technical data type. E.g. 'int'.<br></td>
	</tr>
	<tr>
		<td style="width:150px">temporal</td>
		<td style="width:50px">bool</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">false</td>
		<td>Whether this feature is time dependent and can have different values when measured       on different times (e.g. weight, temporal=true) or generally only measured once (e.g. birth date,       temporal=false).<br></td>
	</tr>
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column identifier should unique.</td></tr>
</table>
</p>


<h3 id="category_entity">File: category.txt</h3> 
<p>Contents:<br> Category is partOf ObservableFeature to define
				categories for an
				ObservableFeature, such as the categorical
				answer codes that are often used in Questionaires. 
				For example the ObservableFeature 'sex' has
				{code_string = 1, label=male} and {code_string
				= 2, label=female}.
				Category can be linked to well-defined
				ontology terms via the
				ontologyReference. Category
				extends
				ObservationElement such that it
				can be referenced by
				ObservedValue.value.
				The Category class maps to
				METABASE::Category
			
.</p>
<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">identifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>user supplied or automatically assigned (using a decorator) unique and short identifier, e.g. MA1234.<br></td>
	</tr>
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>human readible name, not necessary unique.<br></td>
	</tr>
	<tr>
		<td style="width:150px">description</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>(Optional) Rudimentary meta data about the observable feature. Use of ontology       terms references to establish unambigious descriptions is recommended.<br></td>
	</tr>
	<tr>
		<td style="width:150px">observablefeature_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>The Measurement these permitted values are part of..
		This xref uses {observablefeature_identifier} to find related elements in file <a href="#observableFeature_entity">observableFeature.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">valuecode</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>The value used to store this category in ObservedValue. For example '1', '2'.<br></td>
	</tr>
	<tr>
		<td style="width:150px">ismissing</td>
		<td style="width:50px">bool</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">false</td>
		<td>whether this value should be treated as missing value.<br></td>
	</tr>
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column identifier should unique.</td></tr>
</table>
</p>


<h3 id="protocol_entity">File: protocol.txt</h3> 
<p>Contents:<br> 
				The Protocol class defines parameterizable descriptions
				of
				(analysis)methods. Examples of protocols are: Questionaires, SOPs,
				Assay platforms, Statistical analyses, etc.
				Each protocol has a
				unique identifier.
				Protocol has an association to OntologyTerm to
				represent the type of
				protocol.
			
</p>
<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">identifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>user supplied or automatically assigned (using a decorator) unique and short identifier, e.g. MA1234.<br></td>
	</tr>
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>human readible name, not necessary unique.<br></td>
	</tr>
	<tr>
		<td style="width:150px">description</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>(Optional) Rudimentary meta data about the observable feature. Use of ontology       terms references to establish unambigious descriptions is recommended.<br></td>
	</tr>
	<tr>
		<td style="width:150px">protocoltype_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>classification of protocol.
		This xref uses {protocoltype_identifier} to find related elements in file <a href="#ontologyTerm_entity">ontologyTerm.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">subprotocols_Identifier<br></td>
		<td style="width:50px">mref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Subprotocols of this protocol.
		This mref uses {subprotocols_identifier} to find related elements in file <a href="#protocol_entity">protocol.txt</a> based on unique column {identifier}.
		. More than one reference can be added separated by '|', for example: ref1|ref2|ref3.</td>
	</tr>	
	<tr>
		<td style="width:150px">features_Identifier<br></td>
		<td style="width:50px">mref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>parameters (in/out) that are used or produced by this protocol..
		This mref uses {features_identifier} to find related elements in file <a href="#observableFeature_entity">observableFeature.txt</a> based on unique column {identifier}.
		. More than one reference can be added separated by '|', for example: ref1|ref2|ref3.</td>
	</tr>	
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column identifier should unique.</td></tr>
</table>
</p>


<h3 id="dataSet_entity">File: dataset.txt</h3> 
<p>Contents:<br> Container for one or more observations that are measured
				using the same protocol and by the same performer(s). The dataset
				may be a file (having the same
				identifier) but in most cases it is a
				data table consisting of rows (Observation).

				This entity replaces
				ProtocolApplication.
			
</p>
<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">identifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>user supplied or automatically assigned (using a decorator) unique and short identifier, e.g. MA1234.<br></td>
	</tr>
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>human readible name, not necessary unique.<br></td>
	</tr>
	<tr>
		<td style="width:150px">description</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>(Optional) Rudimentary meta data about the observable feature. Use of ontology       terms references to establish unambigious descriptions is recommended.<br></td>
	</tr>
	<tr>
		<td style="width:150px">protocolused_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Reference to the protocol that is being used (if available).
		This xref uses {protocolused_identifier} to find related elements in file <a href="#protocol_entity">protocol.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">starttime</td>
		<td style="width:50px">datetime</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">today</td>
		<td>time when the protocol started.<br></td>
	</tr>
	<tr>
		<td style="width:150px">endtime</td>
		<td style="width:50px">datetime</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">today</td>
		<td>(Optional) time when the protocol ended.<br></td>
	</tr>
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column identifier should unique.</td></tr>
</table>
</p>


<h3 id="observationSet_entity">File: observationset.txt</h3> 
<p>Contents:<br> In practice: Observation is one row within a DataSet.
			
</p>
<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">partofdataset_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>DataSet this ValueSet is part of..
		This xref uses {partofdataset_identifier} to find related elements in file <a href="#dataSet_entity">dataSet.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">target_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>References the target for which this data was recorded. For example 'individual1'..
		This xref uses {target_identifier} to find related elements in file <a href="#characteristic_entity">characteristic.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">time</td>
		<td style="width:50px">datetime</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Time of this observationSet.<br></td>
	</tr>
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Contraint: values in the combined columns (partofdataset, target, time) should be unique.</td></tr>
</table>
</p>


<h3 id="observedValue_entity">File: observedvalue.txt</h3> 
<p>Contents:<br> 
				Generic storage of values as part of one observation
				event. Values are atomatic observations,
				e.g.,
				length (feature) of
				individual 1 (valueset.target)
				= 179cm (value).
				Values can also be
				qualified by some characteristic,
				e.g., QTL
				p-value (feature)
				between
				phenotype 'leaf count'
				(characteristic) and
				marker 'PVV4'
				(valueset.target) =
				0.1^10+3 (value).
			
</p>
<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">observationset_id<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Reference to the observation. For example a particular patient visit or the application of a microarray or the calculation of a QTL model.
		This xref uses {observationset_id} to find related elements in file <a href="#observationSet_entity">observationSet.txt</a> based on unique column {id}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">feature_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>References the ObservableFeature that this observation was made on. For example 'probe123'..
		This xref uses {feature_identifier} to find related elements in file <a href="#observableFeature_entity">observableFeature.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">characteristic_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Is brother of [characteristic], or Average of [height].
		This xref uses {characteristic_identifier} to find related elements in file <a href="#characteristic_entity">characteristic.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">value</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>The value observed.<br></td>
	</tr>
	<!-- all optional fields -->
	<!-- all the uniques -->
</table>
</p>


<h3 id="species_entity">File: species.txt</h3> 
<p>Contents:<br> Ontology terms for species. E.g. Arabidopsis thaliana.
				DISCUSSION: should we avoid subclasses of OntologyTerm and instead
				make a 'tag' filter on terms so we can make pulldowns context
				dependent (e.g. to only show particular subqueries of ontologies).
			
</p>
<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">identifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>user supplied or automatically assigned (using a decorator) unique and short identifier, e.g. MA1234.<br></td>
	</tr>
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>human readible name, not necessary unique.<br></td>
	</tr>
	<tr>
		<td style="width:150px">ontology_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>(Optional) The source ontology or controlled vocabulary list that ontology terms have been obtained from..
		This xref uses {ontology_identifier} to find related elements in file <a href="#ontology_entity">ontology.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">termaccession</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>(Optional) The accession number assigned to the ontology term in its source ontology. If empty it is assumed to be a locally defined term.<br></td>
	</tr>
	<tr>
		<td style="width:150px">definition</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>(Optional) The definition of the term.<br></td>
	</tr>
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column identifier should unique.</td></tr>
<tr><td colspan="5">Contraint: values in the combined columns (ontology, termaccession) should be unique.</td></tr>
</table>
</p>


<h3 id="individual_entity">File: individual.txt</h3> 
<p>Contents:<br>  The Individuals class defines the subjects that are
				used
				as observation target. The Individual class maps to
				XGAP:Individual
				and PaGE:Individual. Groups of individuals can be
				defined via
				Panel.
			
</p>
<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">identifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>user supplied or automatically assigned (using a decorator) unique and short identifier, e.g. MA1234.<br></td>
	</tr>
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>human readible name, not necessary unique.<br></td>
	</tr>
	<tr>
		<td style="width:150px">description</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>(Optional) Rudimentary meta data about the observable feature. Use of ontology       terms references to establish unambigious descriptions is recommended.<br></td>
	</tr>
	<tr>
		<td style="width:150px">mother_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Refers to the mother of the individual..
		This xref uses {mother_identifier} to find related elements in file <a href="#individual_entity">individual.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">father_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Refers to the father of the individual..
		This xref uses {father_identifier} to find related elements in file <a href="#individual_entity">individual.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column identifier should unique.</td></tr>
</table>
</p>


<h3 id="panel_entity">File: panel.txt</h3> 
<p>Contents:<br> The Panel class defines groups of individuals based on
				cohort design, case/controls, families, etc. For instance:
				'LifeLines
				cohort', 'middle aged man', 'recombinant mouse inbred Line
				dba x b6'
				or 'Smith family'. A Panel can act as a single
				ObservationTarget.
				For example: average height (Measurement) in the
				LifeLines cohort
				(Panel) is 174cm (ObservedValue). The Panel class
				maps to XGAP:Strain and PaGE:Panel
				classes. In METABASE this is
				assumed there is one panel per study.
			
</p>
<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">identifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>user supplied or automatically assigned (using a decorator) unique and short identifier, e.g. MA1234.<br></td>
	</tr>
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>human readible name, not necessary unique.<br></td>
	</tr>
	<tr>
		<td style="width:150px">description</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>(Optional) Rudimentary meta data about the observable feature. Use of ontology       terms references to establish unambigious descriptions is recommended.<br></td>
	</tr>
	<tr>
		<td style="width:150px">paneltype_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Indicate the type of Panel (example: Sample panel, AssayedPanel, Natural=wild type, Parental=parents of a cross, F1=First generation of cross, RCC=Recombinant congenic, CSS=chromosome substitution).
		This xref uses {paneltype_identifier} to find related elements in file <a href="#ontologyTerm_entity">ontologyTerm.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">numberofindividuals</td>
		<td style="width:50px">int</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>NumberOfIndividuals.<br></td>
	</tr>
	<tr>
		<td style="width:150px">species_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>The species this panel is an instance of/part of/extracted from..
		This xref uses {species_identifier} to find related elements in file <a href="#species_entity">species.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">individuals_Identifier<br></td>
		<td style="width:50px">mref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>The list of individuals in this panel.
		This mref uses {individuals_identifier} to find related elements in file <a href="#individual_entity">individual.txt</a> based on unique column {identifier}.
		. More than one reference can be added separated by '|', for example: ref1|ref2|ref3.</td>
	</tr>	
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column identifier should unique.</td></tr>
</table>
</p>


<h3 id="panelSource_entity">File: panelsource.txt</h3> 
<p>Contents:<br> PanelSources is partOf Panel to define how panels are
				related panels, founder panels,
				such as overlap,
				selection criteria,
				getting assayed panel from a
				sample panel, etc.
			
</p>
<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">currentpanel_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Panel for which these sources are defined..
		This xref uses {currentpanel_identifier} to find related elements in file <a href="#panel_entity">panel.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">sourcepanel_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Source that contributed individuals to current panel.
		This xref uses {sourcepanel_identifier} to find related elements in file <a href="#panel_entity">panel.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">numberofindividuals</td>
		<td style="width:50px">int</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Number of individuals lifted over from this source.<br></td>
	</tr>
	<tr>
		<td style="width:150px">selectioncriteria</td>
		<td style="width:50px">text</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Inclusion/exclusion criteria used to select these individuals from source into current panel.<br></td>
	</tr>
	<!-- all optional fields -->
	<!-- all the uniques -->
</table>
</p>


<h3 id="ontology_entity">File: ontology.txt</h3> 
<p>Contents:<br>  Ontology defines a reference to an ontology or
				controlled vocabulary from which well-defined and stable (ontology)
				terms can be obtained. Each Ontology should have a unique identifer,
				for instance: Gene Ontology, Mammalian Phenotype, Human Phenotype
				Ontology, Unified Medical Language System, Medical Subject Headings,
				etc. Also a abbreviation is required, for instance: GO, MP, HPO,
				UMLS, MeSH, etc. Use of existing ontologies/vocabularies is
				recommended to harmonize phenotypic feature and value descriptions.
				But one can also create a 'local' Ontology. The Ontology class maps
				to FuGE::Ontology, MAGE-TAB::TermSourceREF.
			
</p>
<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">identifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>user supplied or automatically assigned (using a decorator) unique and short identifier, e.g. MA1234.<br></td>
	</tr>
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>human readible name, not necessary unique.<br></td>
	</tr>
	<tr>
		<td style="width:150px">ontologyaccession</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>A accession that uniquely identifies the ontology (typically an acronym). E.g. GO, MeSH, HPO.<br></td>
	</tr>
	<tr>
		<td style="width:150px">ontologyuri</td>
		<td style="width:50px">hyperlink</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>(Optional) A URI that references the location of the ontology.<br></td>
	</tr>
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column identifier should unique.</td></tr>
</table>
</p>


<h3 id="ontologyTerm_entity">File: ontologyterm.txt</h3> 
<p>Contents:<br> 
				OntologyTerm defines a single entry (term) from an
				ontology or a controlled vocabulary (defined by Ontology). The
				identifier is the ontology term is unique. E.g. 'NCI:Antigen Gene'.
				Other data entities can reference to this OntologyTerm to harmonize
				naming of concepts. If no suitable ontology term exists then one can
				define new terms locally (in which case there is no formal accession
				for the term limiting its use for cross-Investigation queries).
			
</p>
<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">identifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>user supplied or automatically assigned (using a decorator) unique and short identifier, e.g. MA1234.<br></td>
	</tr>
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>human readible name, not necessary unique.<br></td>
	</tr>
	<tr>
		<td style="width:150px">ontology_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>(Optional) The source ontology or controlled vocabulary list that ontology terms have been obtained from..
		This xref uses {ontology_identifier} to find related elements in file <a href="#ontology_entity">ontology.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">termaccession</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>(Optional) The accession number assigned to the ontology term in its source ontology. If empty it is assumed to be a locally defined term.<br></td>
	</tr>
	<tr>
		<td style="width:150px">definition</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>(Optional) The definition of the term.<br></td>
	</tr>
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column identifier should unique.</td></tr>
<tr><td colspan="5">Contraint: values in the combined columns (ontology, termaccession) should be unique.</td></tr>
</table>
</p>


<h3 id="accession_entity">File: accession.txt</h3> 
<p>Contents:<br> 
				An external identifier for an annotation. For example:
				name='R13H8.1', ontology='ensembl' or name='WBgene00000912',
				ontology='wormbase'.
			
</p>
<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">identifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>user supplied or automatically assigned (using a decorator) unique and short identifier, e.g. MA1234.<br></td>
	</tr>
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>human readible name, not necessary unique.<br></td>
	</tr>
	<tr>
		<td style="width:150px">ontology_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>(Optional) The source ontology or controlled vocabulary list that ontology terms have been obtained from..
		This xref uses {ontology_identifier} to find related elements in file <a href="#ontology_entity">ontology.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">termaccession</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>(Optional) The accession number assigned to the ontology term in its source ontology. If empty it is assumed to be a locally defined term.<br></td>
	</tr>
	<tr>
		<td style="width:150px">definition</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>(Optional) The definition of the term.<br></td>
	</tr>
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column identifier should unique.</td></tr>
<tr><td colspan="5">Contraint: values in the combined columns (ontology, termaccession) should be unique.</td></tr>
</table>
</p>


<h3 id="genome_entity">File: genome.txt</h3> 

<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">identifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>user supplied or automatically assigned (using a decorator) unique and short identifier, e.g. MA1234.<br></td>
	</tr>
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>human readible name, not necessary unique.<br></td>
	</tr>
	<tr>
		<td style="width:150px">description</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>(Optional) Rudimentary meta data about the observable feature. Use of ontology       terms references to establish unambigious descriptions is recommended.<br></td>
	</tr>
	<tr>
		<td style="width:150px">residues</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>A sequence of alphabetic characters representing biological residues (nucleic acids, amino acids). This column does not need to be manifested for all features; it is optional for features such as exons where the residues can be derived from the featureloc. It is recommended that the value for this column be manifested for features which may may non-contiguous sublocations (e.g. transcripts), since derivation at query time is non-trivial. For expressed sequence, the DNA sequence should be used rather than the RNA sequence.<br></td>
	</tr>
	<tr>
		<td style="width:150px">seqlen</td>
		<td style="width:50px">int</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>The length of the residue feature. See column:residues. This column is partially redundant with the residues column, and also with featureloc. This column is required because the location may be unknown and the residue sequence may not be manifested, yet it may be desirable to store and query the length of the feature. The seqlen should always be manifested where the length of the sequence is known.<br></td>
	</tr>
	<tr>
		<td style="width:150px">species_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>species this genome belongs to..
		This xref uses {species_identifier} to find related elements in file <a href="#species_entity">species.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column identifier should unique.</td></tr>
</table>
</p>


<h3 id="chromosome_entity">File: chromosome.txt</h3> 
<p>Contents:<br> Example: b37:chr1.
</p>
<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">identifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>user supplied or automatically assigned (using a decorator) unique and short identifier, e.g. MA1234.<br></td>
	</tr>
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>human readible name, not necessary unique.<br></td>
	</tr>
	<tr>
		<td style="width:150px">description</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>(Optional) Rudimentary meta data about the observable feature. Use of ontology       terms references to establish unambigious descriptions is recommended.<br></td>
	</tr>
	<tr>
		<td style="width:150px">residues</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>A sequence of alphabetic characters representing biological residues (nucleic acids, amino acids). This column does not need to be manifested for all features; it is optional for features such as exons where the residues can be derived from the featureloc. It is recommended that the value for this column be manifested for features which may may non-contiguous sublocations (e.g. transcripts), since derivation at query time is non-trivial. For expressed sequence, the DNA sequence should be used rather than the RNA sequence.<br></td>
	</tr>
	<tr>
		<td style="width:150px">seqlen</td>
		<td style="width:50px">int</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>The length of the residue feature. See column:residues. This column is partially redundant with the residues column, and also with featureloc. This column is required because the location may be unknown and the residue sequence may not be manifested, yet it may be desirable to store and query the length of the feature. The seqlen should always be manifested where the length of the sequence is known.<br></td>
	</tr>
	<tr>
		<td style="width:150px">genome_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>genome.
		This xref uses {genome_identifier} to find related elements in file <a href="#genome_entity">genome.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">ordernr</td>
		<td style="width:50px">int</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>orderNr.<br></td>
	</tr>
	<tr>
		<td style="width:150px">isautosomal</td>
		<td style="width:50px">bool</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Is 'yes' when number of chromosomes is equal in male and female individuals, i.e., if not a sex chromosome.<br></td>
	</tr>
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column identifier should unique.</td></tr>
</table>
</p>


<h3 id="gene_entity">File: gene.txt</h3> 
<p>Contents:<br> Serves as a view on SequenceCharacteristics that are
				genes / cdna
			
.</p>
<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">identifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>user supplied or automatically assigned (using a decorator) unique and short identifier, e.g. MA1234.<br></td>
	</tr>
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>human readible name, not necessary unique.<br></td>
	</tr>
	<tr>
		<td style="width:150px">description</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>(Optional) Rudimentary meta data about the observable feature. Use of ontology       terms references to establish unambigious descriptions is recommended.<br></td>
	</tr>
	<tr>
		<td style="width:150px">gdna_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>The genome this element lies on..
		This xref uses {gdna_identifier} to find related elements in file <a href="#chromosome_entity">chromosome.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">gdna_start</td>
		<td style="width:50px">int</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Start position on genomic sequence.<br></td>
	</tr>
	<tr>
		<td style="width:150px">gdna_end</td>
		<td style="width:50px">int</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>End position on genomic sequence.<br></td>
	</tr>
	<tr>
		<td style="width:150px">residues</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>A sequence of alphabetic characters representing biological residues (nucleic acids, amino acids). This column does not need to be manifested for all features; it is optional for features such as exons where the residues can be derived from the featureloc. It is recommended that the value for this column be manifested for features which may may non-contiguous sublocations (e.g. transcripts), since derivation at query time is non-trivial. For expressed sequence, the DNA sequence should be used rather than the RNA sequence.<br></td>
	</tr>
	<tr>
		<td style="width:150px">seqlen</td>
		<td style="width:50px">int</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>The length of the residue feature. See column:residues. This column is partially redundant with the residues column, and also with featureloc. This column is required because the location may be unknown and the residue sequence may not be manifested, yet it may be desirable to store and query the length of the feature. The seqlen should always be manifested where the length of the sequence is known.<br></td>
	</tr>
	<tr>
		<td style="width:150px">strand</td>
		<td style="width:50px">enum</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>The orientation/directionality of the location. Should be 0, -1 or +1.<br></td>
	</tr>
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column identifier should unique.</td></tr>
</table>
</p>


<h3 id="protein_entity">File: protein.txt</h3> 
<p>Contents:<br> Serves as a view on SequencesCharacteristics that are
				proteins / aa.
			
</p>
<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">identifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>user supplied or automatically assigned (using a decorator) unique and short identifier, e.g. MA1234.<br></td>
	</tr>
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>human readible name, not necessary unique.<br></td>
	</tr>
	<tr>
		<td style="width:150px">description</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>(Optional) Rudimentary meta data about the observable feature. Use of ontology       terms references to establish unambigious descriptions is recommended.<br></td>
	</tr>
	<tr>
		<td style="width:150px">cdna_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>The gene this element lies on..
		This xref uses {cdna_identifier} to find related elements in file <a href="#gene_entity">gene.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">cdna_start</td>
		<td style="width:50px">int</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Start position on cDNA sequence.<br></td>
	</tr>
	<tr>
		<td style="width:150px">cdna_end</td>
		<td style="width:50px">int</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>End position on cDNA sequence.<br></td>
	</tr>
	<tr>
		<td style="width:150px">residues</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>A sequence of alphabetic characters representing biological residues (nucleic acids, amino acids). This column does not need to be manifested for all features; it is optional for features such as exons where the residues can be derived from the featureloc. It is recommended that the value for this column be manifested for features which may may non-contiguous sublocations (e.g. transcripts), since derivation at query time is non-trivial. For expressed sequence, the DNA sequence should be used rather than the RNA sequence.<br></td>
	</tr>
	<tr>
		<td style="width:150px">seqlen</td>
		<td style="width:50px">int</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>The length of the residue feature. See column:residues. This column is partially redundant with the residues column, and also with featureloc. This column is required because the location may be unknown and the residue sequence may not be manifested, yet it may be desirable to store and query the length of the feature. The seqlen should always be manifested where the length of the sequence is known.<br></td>
	</tr>
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column identifier should unique.</td></tr>
</table>
</p>


<h3 id="proteinDomain_entity">File: proteindomain.txt</h3> 
<p>Contents:<br> Serves as a view on SequenceCharacteristics that are
				protein domains.
			
</p>
<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">identifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>user supplied or automatically assigned (using a decorator) unique and short identifier, e.g. MA1234.<br></td>
	</tr>
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>human readible name, not necessary unique.<br></td>
	</tr>
	<tr>
		<td style="width:150px">description</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>(Optional) Rudimentary meta data about the observable feature. Use of ontology       terms references to establish unambigious descriptions is recommended.<br></td>
	</tr>
	<tr>
		<td style="width:150px">cdna_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>The gene this element lies on..
		This xref uses {cdna_identifier} to find related elements in file <a href="#gene_entity">gene.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">cdna_start</td>
		<td style="width:50px">int</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Start position on cDNA sequence.<br></td>
	</tr>
	<tr>
		<td style="width:150px">cdna_end</td>
		<td style="width:50px">int</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>End position on cDNA sequence.<br></td>
	</tr>
	<tr>
		<td style="width:150px">gdna_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>The genome this element lies on..
		This xref uses {gdna_identifier} to find related elements in file <a href="#chromosome_entity">chromosome.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">gdna_start</td>
		<td style="width:50px">int</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Start position on genomic sequence.<br></td>
	</tr>
	<tr>
		<td style="width:150px">gdna_end</td>
		<td style="width:50px">int</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>End position on genomic sequence.<br></td>
	</tr>
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column identifier should unique.</td></tr>
</table>
</p>


<h3 id="exon_entity">File: exon.txt</h3> 
<p>Contents:<br> Serves as a view on SequenceCharacteristics that are
				exons or introns.
			
</p>
<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">identifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>user supplied or automatically assigned (using a decorator) unique and short identifier, e.g. MA1234.<br></td>
	</tr>
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>human readible name, not necessary unique.<br></td>
	</tr>
	<tr>
		<td style="width:150px">description</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>(Optional) Rudimentary meta data about the observable feature. Use of ontology       terms references to establish unambigious descriptions is recommended.<br></td>
	</tr>
	<tr>
		<td style="width:150px">cdna_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>The gene this element lies on..
		This xref uses {cdna_identifier} to find related elements in file <a href="#gene_entity">gene.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">cdna_start</td>
		<td style="width:50px">int</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Start position on cDNA sequence.<br></td>
	</tr>
	<tr>
		<td style="width:150px">cdna_end</td>
		<td style="width:50px">int</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>End position on cDNA sequence.<br></td>
	</tr>
	<tr>
		<td style="width:150px">gdna_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>The genome this element lies on..
		This xref uses {gdna_identifier} to find related elements in file <a href="#chromosome_entity">chromosome.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">gdna_start</td>
		<td style="width:50px">int</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Start position on genomic sequence.<br></td>
	</tr>
	<tr>
		<td style="width:150px">gdna_end</td>
		<td style="width:50px">int</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>End position on genomic sequence.<br></td>
	</tr>
	<tr>
		<td style="width:150px">isintron</td>
		<td style="width:50px">bool</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Is this exon actually an intron?.<br></td>
	</tr>
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column identifier should unique.</td></tr>
</table>
</p>


<h3 id="variant_entity">File: variant.txt</h3> 
<p>Contents:<br> Convenient entity to define in one place variants on
				gdna, cdna and aa level.
			
</p>
<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">identifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>user supplied or automatically assigned (using a decorator) unique and short identifier, e.g. MA1234.<br></td>
	</tr>
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>human readible name, not necessary unique.<br></td>
	</tr>
	<tr>
		<td style="width:150px">description</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>(Optional) Rudimentary meta data about the observable feature. Use of ontology       terms references to establish unambigious descriptions is recommended.<br></td>
	</tr>
	<tr>
		<td style="width:150px">gdna_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>The genome this element lies on..
		This xref uses {gdna_identifier} to find related elements in file <a href="#chromosome_entity">chromosome.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">gdna_start</td>
		<td style="width:50px">int</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Start position on genomic sequence.<br></td>
	</tr>
	<tr>
		<td style="width:150px">gdna_end</td>
		<td style="width:50px">int</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>End position on genomic sequence.<br></td>
	</tr>
	<tr>
		<td style="width:150px">cdna_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>The gene this element lies on..
		This xref uses {cdna_identifier} to find related elements in file <a href="#gene_entity">gene.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">cdna_start</td>
		<td style="width:50px">int</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Start position on cDNA sequence.<br></td>
	</tr>
	<tr>
		<td style="width:150px">cdna_end</td>
		<td style="width:50px">int</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>End position on cDNA sequence.<br></td>
	</tr>
	<tr>
		<td style="width:150px">aa_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>The protein sequence this element lies on..
		This xref uses {aa_identifier} to find related elements in file <a href="#protein_entity">protein.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">aa_start</td>
		<td style="width:50px">int</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Start position on amino acid sequence.<br></td>
	</tr>
	<tr>
		<td style="width:150px">aa_end</td>
		<td style="width:50px">int</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>End position on amino acid sequence.<br></td>
	</tr>
	<tr>
		<td style="width:150px">gdna_notation</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>gDNA notation of the variant, e.g. g.1234567C>T.<br></td>
	</tr>
	<tr>
		<td style="width:150px">cdna_notation</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>cDNA notation of the variant, e.g. c.123C>T.<br></td>
	</tr>
	<tr>
		<td style="width:150px">aa_notation</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Aa notation of the variant, e.g. p.Ser123ArgfsX12.<br></td>
	</tr>
	<tr>
		<td style="width:150px">varianttype_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Type of the variant..
		This xref uses {varianttype_identifier} to find related elements in file <a href="#ontologyTerm_entity">ontologyTerm.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column identifier should unique.</td></tr>
</table>
</p>


<h3 id="study_entity">File: study.txt</h3> 
<p>Contents:<br> Investigation defines self-contained units of study. For
				example: Framingham study. Optionally a description and an accession
				to a data source can be provided. Each Investigation has a unique
				identifier.
			
</p>
<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">identifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>user supplied or automatically assigned (using a decorator) unique and short identifier, e.g. MA1234.<br></td>
	</tr>
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>human readible name, not necessary unique.<br></td>
	</tr>
	<tr>
		<td style="width:150px">description</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>(Optional)      Rudimentary meta data about the Investigation.<br></td>
	</tr>
	<tr>
		<td style="width:150px">startdate</td>
		<td style="width:50px">datetime</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">today</td>
		<td>The start point of the study.<br></td>
	</tr>
	<tr>
		<td style="width:150px">updatedate</td>
		<td style="width:50px">datetime</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">today</td>
		<td>Last time the investigation was modified.<br></td>
	</tr>
	<tr>
		<td style="width:150px">enddate</td>
		<td style="width:50px">datetime</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>The end point of the study.<br></td>
	</tr>
	<tr>
		<td style="width:150px">contact_Name<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Primary contact person for this study.
		This xref uses {contact_name} to find related elements in file <a href="#person_entity">person.txt</a> based on unique column {name}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">partofinvestigation_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>xref to the investigation the study is part of.
		This xref uses {partofinvestigation_identifier} to find related elements in file <a href="#investigation_entity">investigation.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column identifier should unique.</td></tr>
</table>
</p>


<h3 id="experiment_entity">File: experiment.txt</h3> 

<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">identifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>user supplied or automatically assigned (using a decorator) unique and short identifier, e.g. MA1234.<br></td>
	</tr>
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>human readible name, not necessary unique.<br></td>
	</tr>
	<tr>
		<td style="width:150px">study_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Part of Study..
		This xref uses {study_identifier} to find related elements in file <a href="#study_entity">study.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">design</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Design.<br></td>
	</tr>
	<tr>
		<td style="width:150px">experimenttype_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Experiment type. E.g. 'case-control'..
		This xref uses {experimenttype_identifier} to find related elements in file <a href="#ontologyTerm_entity">ontologyTerm.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">totalmarkerstested</td>
		<td style="width:50px">int</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Total markers tested.<br></td>
	</tr>
	<tr>
		<td style="width:150px">totalmarkersimported</td>
		<td style="width:50px">int</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Total markers imported.<br></td>
	</tr>
	<tr>
		<td style="width:150px">objective</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Objective.<br></td>
	</tr>
	<tr>
		<td style="width:150px">outcome</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Outcome.<br></td>
	</tr>
	<tr>
		<td style="width:150px">comments</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Comments.<br></td>
	</tr>
	<tr>
		<td style="width:150px">individualdatastatement</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">Access to individual-level data must be made to the study authors</td>
		<td>Individual data statement.<br></td>
	</tr>
	<tr>
		<td style="width:150px">timecreated</td>
		<td style="width:50px">datetime</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Time created.<br></td>
	</tr>
	<tr>
		<td style="width:150px">assayedpanels_Identifier<br></td>
		<td style="width:50px">mref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Assayed panels.
		This mref uses {assayedpanels_identifier} to find related elements in file <a href="#panel_entity">panel.txt</a> based on unique column {identifier}.
		. More than one reference can be added separated by '|', for example: ref1|ref2|ref3.</td>
	</tr>	
	<tr>
		<td style="width:150px">datasets_Identifier<br></td>
		<td style="width:50px">mref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>DataSets that were input/output of this experiment..
		This mref uses {datasets_identifier} to find related elements in file <a href="#dataSet_entity">dataSet.txt</a> based on unique column {identifier}.
		. More than one reference can be added separated by '|', for example: ref1|ref2|ref3.</td>
	</tr>	
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column identifier should unique.</td></tr>
</table>
</p>


<h3 id="institute_entity">File: institute.txt</h3> 
<p>Contents:<br>  A contact is either a person or an organization. Copied
				from FuGE::Contact.
			
</p>
<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>name.<br></td>
	</tr>
	<tr>
		<td style="width:150px">address</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>The address of the Contact.<br></td>
	</tr>
	<tr>
		<td style="width:150px">phone</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>The telephone number of the Contact including the suitable area codes.<br></td>
	</tr>
	<tr>
		<td style="width:150px">city</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Added from the old definition of MolgenisUser. City of this contact.<br></td>
	</tr>
	<tr>
		<td style="width:150px">country</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Added from the old definition of MolgenisUser. Country of this contact.<br></td>
	</tr>
	<tr>
		<td style="width:150px">fax</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>The fax number of the Contact.<br></td>
	</tr>
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column name should unique.</td></tr>
</table>
</p>


<h3 id="person_entity">File: person.txt</h3> 
<p>Contents:<br> 

				Person represents one or more people involved with an Investigation.
				This may include authors on a paper, lab personnel or PIs. Person
				has last name, firstname, mid initial, address, contact and email. A
				Person role is included to represent how a Person is involved with
				an investigation. For submission to repository purposes an allowed
				value is 'submitter' and the term is present in the MGED Ontology,
				an alternative use could represent job title. An Example from
				ArrayExpress is E-MTAB-506
				<a href="ftp://ftp.ebi.ac.uk/pub/databases/microarray/data/experiment/TABM/E-TABM-506/E-TABM-506.idf.txt">
					ftp://ftp.ebi.ac.uk/pub/databases/microarray/data/experiment/TABM/E-TABM-506/E-TABM-506.idf.txt.
				</a>
				.
    <!-- From MIBBI : Name should either be of the person or the institutional role holding responsibility for the associated (meta)data. -->
    <br/>
				The FUGE equivalent to Person is FuGE::Person.
			
</p>
<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Name.<br></td>
	</tr>
	<tr>
		<td style="width:150px">title</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>An academic title, e.g. Prof.dr, PhD.<br></td>
	</tr>
	<tr>
		<td style="width:150px">firstname</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>First Name.<br></td>
	</tr>
	<tr>
		<td style="width:150px">midinitials</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Mid Initials.<br></td>
	</tr>
	<tr>
		<td style="width:150px">lastname</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Last Name.<br></td>
	</tr>
	<tr>
		<td style="width:150px">email</td>
		<td style="width:50px">email</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Email.<br></td>
	</tr>
	<tr>
		<td style="width:150px">phone</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>The telephone number of the Contact including the suitable area codes.<br></td>
	</tr>
	<tr>
		<td style="width:150px">primaryaffilation_name<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Affliations.
		This xref uses {primaryaffilation_name} to find related elements in file <a href="#institute_entity">institute.txt</a> based on unique column {name}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">affiliateinstitutions_name<br></td>
		<td style="width:50px">mref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Affliated Institutes.
		This mref uses {affiliateinstitutions_name} to find related elements in file <a href="#institute_entity">institute.txt</a> based on unique column {name}.
		. More than one reference can be added separated by '|', for example: ref1|ref2|ref3.</td>
	</tr>	
	<tr>
		<td style="width:150px">orcidpersonreference_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>OrcidPersonReference.
		This xref uses {orcidpersonreference_identifier} to find related elements in file <a href="#ontologyTerm_entity">ontologyTerm.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column name should unique.</td></tr>
<tr><td colspan="5">Contraint: values in the combined columns (firstname, midinitials, lastname) should be unique.</td></tr>
</table>
</p>


<h3 id="citation_entity">File: citation.txt</h3> 
<p>Contents:<br> 

				Citation of a document or data set. Publications have
				attributes of
				publications Authors and also DOI and Pubmed
				identifiers (when these
				are available). These are represented as
				OntologyTerms as in the
				MAGE-TAB model all 'xrefs' (cross
				references) for ontologies and
				accession numbers are handled
				generically. An example of a
				publication is available in an IDF file
				from ArrayExpress is
				experiment E-MTAB-506
				<a href="ftp://ftp.ebi.ac.uk/pub/databases/microarray/data/experiment/TABM/E-TABM-506/E-TABM-506.idf.txt">ftp://ftp.ebi.ac.uk/pub/databases/microarray/data/experiment/TABM/E-TABM-506/E-TABM-506.idf.txt
				</a>
				.
				<br/>
				The FuGE equivalent to Publication is FuGE::Bibliographic Reference.
			
</p>
<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">identifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>user supplied or automatically assigned (using a decorator) unique and short identifier, e.g. MA1234.<br></td>
	</tr>
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>human readible name, not necessary unique.<br></td>
	</tr>
	<tr>
		<td style="width:150px">pubmedid</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Pubmed ID.<br></td>
	</tr>
	<tr>
		<td style="width:150px">doi</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Publication DOI.<br></td>
	</tr>
	<tr>
		<td style="width:150px">ontologyterms_Identifier<br></td>
		<td style="width:50px">mref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>ontology terms such as MeSH.
		This mref uses {ontologyterms_identifier} to find related elements in file <a href="#ontologyTerm_entity">ontologyTerm.txt</a> based on unique column {identifier}.
		. More than one reference can be added separated by '|', for example: ref1|ref2|ref3.</td>
	</tr>	
	<tr>
		<td style="width:150px">authorlist</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>The names of the authors of the publication.<br></td>
	</tr>
	<tr>
		<td style="width:150px">title</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>The title of the Publication.<br></td>
	</tr>
	<tr>
		<td style="width:150px">description</td>
		<td style="width:50px">text</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Description.<br></td>
	</tr>
	<tr>
		<td style="width:150px">status_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>The status of the Publication.
		This xref uses {status_identifier} to find related elements in file <a href="#ontologyTerm_entity">ontologyTerm.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column identifier should unique.</td></tr>
<tr><td colspan="5">Constraint: values in column pubmedid should unique.</td></tr>
<tr><td colspan="5">Constraint: values in column doi should unique.</td></tr>
</table>
</p>


<h3 id="contribution_entity">File: contribution.txt</h3> 
<p>Contents:<br> A contribution describes the part a Person has
				contributed to a Contributable item.
			
</p>
<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">identifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>user supplied or automatically assigned (using a decorator) unique and short identifier, e.g. MA1234.<br></td>
	</tr>
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>human readible name, not necessary unique.<br></td>
	</tr>
	<tr>
		<td style="width:150px">researcher_Name<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>The person involved.
		This xref uses {researcher_name} to find related elements in file <a href="#person_entity">person.txt</a> based on unique column {name}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">submission_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>The submission contributed to..
		This xref uses {submission_identifier} to find related elements in file <a href="#submission_entity">submission.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">issubmitter</td>
		<td style="width:50px">enum</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Submitter?.<br></td>
	</tr>
	<tr>
		<td style="width:150px">isauthor</td>
		<td style="width:50px">enum</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Author?.<br></td>
	</tr>
	<tr>
		<td style="width:150px">issource</td>
		<td style="width:50px">enum</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Source?.<br></td>
	</tr>
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column identifier should unique.</td></tr>
</table>
</p>


<h3 id="submission_entity">File: submission.txt</h3> 

<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">identifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>user supplied or automatically assigned (using a decorator) unique and short identifier, e.g. MA1234.<br></td>
	</tr>
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>human readible name, not necessary unique.<br></td>
	</tr>
	<tr>
		<td style="width:150px">timecreated</td>
		<td style="width:50px">datetime</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Time created.<br></td>
	</tr>
	<tr>
		<td style="width:150px">study_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Study.
		This xref uses {study_identifier} to find related elements in file <a href="#study_entity">study.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column identifier should unique.</td></tr>
</table>
</p>


<h3 id="investigation_entity">File: investigation.txt</h3> 

<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">identifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>user supplied or automatically assigned (using a decorator) unique and short identifier, e.g. MA1234.<br></td>
	</tr>
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>human readible name, not necessary unique.<br></td>
	</tr>
	<tr>
		<td style="width:150px">title</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Nice title of Investigation.<br></td>
	</tr>
	<tr>
		<td style="width:150px">shortname</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Shorthand name for layout.<br></td>
	</tr>
	<tr>
		<td style="width:150px">version</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Accession version.<br></td>
	</tr>
	<tr>
		<td style="width:150px">background</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Short piece of information describing why the study is taking place, e.g. risk factors for a population.<br></td>
	</tr>
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column identifier should unique.</td></tr>
</table>
</p>


<h3 id="studyDetails_entity">File: studydetails.txt</h3> 

<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">study_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Study.
		This xref uses {study_identifier} to find related elements in file <a href="#study_entity">study.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">title</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Nice title of the paper.<br></td>
	</tr>
	<tr>
		<td style="width:150px">shortname</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Shorthand name for layout.<br></td>
	</tr>
	<tr>
		<td style="width:150px">studyabstract</td>
		<td style="width:50px">text</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Abstract.<br></td>
	</tr>
	<tr>
		<td style="width:150px">version</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Accession version.<br></td>
	</tr>
	<tr>
		<td style="width:150px">background</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Short piece of information describing why the study is taking place, e.g. risk factors for a population.<br></td>
	</tr>
	<tr>
		<td style="width:150px">objectives</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>What this study aims to achieve.<br></td>
	</tr>
	<tr>
		<td style="width:150px">keyresults</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Noticable results from this study.<br></td>
	</tr>
	<tr>
		<td style="width:150px">conclusions</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Description of the conclusions drawn.<br></td>
	</tr>
	<tr>
		<td style="width:150px">studydesign</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Study design.<br></td>
	</tr>
	<tr>
		<td style="width:150px">studysizereason</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Reason for study size.<br></td>
	</tr>
	<tr>
		<td style="width:150px">studypower</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Study power.<br></td>
	</tr>
	<tr>
		<td style="width:150px">sourcesofbias</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Sources of bias.<br></td>
	</tr>
	<tr>
		<td style="width:150px">limitations</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Limitations.<br></td>
	</tr>
	<tr>
		<td style="width:150px">acknowledgements</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Acknowledgements.<br></td>
	</tr>
	<tr>
		<td style="width:150px">primarycitation_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>primaryCitation.
		This xref uses {primarycitation_identifier} to find related elements in file <a href="#citation_entity">citation.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">othercitations_Identifier<br></td>
		<td style="width:50px">mref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Contact persons for this study.
		This mref uses {othercitations_identifier} to find related elements in file <a href="#citation_entity">citation.txt</a> based on unique column {identifier}.
		. More than one reference can be added separated by '|', for example: ref1|ref2|ref3.</td>
	</tr>	
	<tr>
		<td style="width:150px">accession</td>
		<td style="width:50px">hyperlink</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>(Optional) URI or accession number to indicate source of Study. E.g. arrayexpress:M-EXP-2345.<br></td>
	</tr>
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column study should unique.</td></tr>
</table>
</p>


<h3 id="frequencyCluster_entity">File: frequencycluster.txt</h3> 

<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">identifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>user supplied or automatically assigned (using a decorator) unique and short identifier, e.g. MA1234.<br></td>
	</tr>
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>human readible name, not necessary unique.<br></td>
	</tr>
	<tr>
		<td style="width:150px">description</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>(Optional) Rudimentary meta data about the observable feature. Use of ontology       terms references to establish unambigious descriptions is recommended.<br></td>
	</tr>
	<tr>
		<td style="width:150px">protocolused_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Reference to the protocol that is being used (if available).
		This xref uses {protocolused_identifier} to find related elements in file <a href="#protocol_entity">protocol.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">starttime</td>
		<td style="width:50px">datetime</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">today</td>
		<td>time when the protocol started.<br></td>
	</tr>
	<tr>
		<td style="width:150px">endtime</td>
		<td style="width:50px">datetime</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">today</td>
		<td>(Optional) time when the protocol ended.<br></td>
	</tr>
	<tr>
		<td style="width:150px">dataset_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Result set identifier.
		This xref uses {dataset_identifier} to find related elements in file <a href="#dataSet_entity">dataSet.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">usedmarkerset_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Used marker set ID.
		This xref uses {usedmarkerset_identifier} to find related elements in file <a href="#usedMarkerSet_entity">usedMarkerSet.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">markerid</td>
		<td style="width:50px">int</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Marker ID.<br></td>
	</tr>
	<tr>
		<td style="width:150px">numberofgenotypedsamples</td>
		<td style="width:50px">int</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Number of genotyped samples.<br></td>
	</tr>
	<tr>
		<td style="width:150px">pvaluehwe</td>
		<td style="width:50px">decimal</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>P-value HWE.<br></td>
	</tr>
	<tr>
		<td style="width:150px">unadjustedpvalue</td>
		<td style="width:50px">decimal</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Unadjusted p-value.<br></td>
	</tr>
	<tr>
		<td style="width:150px">oddsratiostatement</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Odds ratio statement.<br></td>
	</tr>
	<tr>
		<td style="width:150px">attributableriskstatement</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Attributable risk statement.<br></td>
	</tr>
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column identifier should unique.</td></tr>
</table>
</p>


<h3 id="genotypeFrequency_entity">File: genotypefrequency.txt</h3> 

<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">identifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>user supplied or automatically assigned (using a decorator) unique and short identifier, e.g. MA1234.<br></td>
	</tr>
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>human readible name, not necessary unique.<br></td>
	</tr>
	<tr>
		<td style="width:150px">description</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>(Optional) Rudimentary meta data about the observable feature. Use of ontology       terms references to establish unambigious descriptions is recommended.<br></td>
	</tr>
	<tr>
		<td style="width:150px">protocolused_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Reference to the protocol that is being used (if available).
		This xref uses {protocolused_identifier} to find related elements in file <a href="#protocol_entity">protocol.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">starttime</td>
		<td style="width:50px">datetime</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">today</td>
		<td>time when the protocol started.<br></td>
	</tr>
	<tr>
		<td style="width:150px">endtime</td>
		<td style="width:50px">datetime</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">today</td>
		<td>(Optional) time when the protocol ended.<br></td>
	</tr>
	<tr>
		<td style="width:150px">frequencycluster_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Frequency cluster ID.
		This xref uses {frequencycluster_identifier} to find related elements in file <a href="#frequencyCluster_entity">frequencyCluster.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">genotypecombo</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Genotype combo.<br></td>
	</tr>
	<tr>
		<td style="width:150px">frequencyasproportion</td>
		<td style="width:50px">decimal</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Frequency as proportion.<br></td>
	</tr>
	<tr>
		<td style="width:150px">numbersampleswithgenotype</td>
		<td style="width:50px">int</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Number of samples with genotype.<br></td>
	</tr>
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column identifier should unique.</td></tr>
</table>
</p>


<h3 id="alleleFrequency_entity">File: allelefrequency.txt</h3> 

<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">identifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>user supplied or automatically assigned (using a decorator) unique and short identifier, e.g. MA1234.<br></td>
	</tr>
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>human readible name, not necessary unique.<br></td>
	</tr>
	<tr>
		<td style="width:150px">description</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>(Optional) Rudimentary meta data about the observable feature. Use of ontology       terms references to establish unambigious descriptions is recommended.<br></td>
	</tr>
	<tr>
		<td style="width:150px">protocolused_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Reference to the protocol that is being used (if available).
		This xref uses {protocolused_identifier} to find related elements in file <a href="#protocol_entity">protocol.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">starttime</td>
		<td style="width:50px">datetime</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">today</td>
		<td>time when the protocol started.<br></td>
	</tr>
	<tr>
		<td style="width:150px">endtime</td>
		<td style="width:50px">datetime</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">today</td>
		<td>(Optional) time when the protocol ended.<br></td>
	</tr>
	<tr>
		<td style="width:150px">frequencycluster_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Frequency cluster ID.
		This xref uses {frequencycluster_identifier} to find related elements in file <a href="#frequencyCluster_entity">frequencyCluster.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">allelecombo</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Allele combo.<br></td>
	</tr>
	<tr>
		<td style="width:150px">frequencyasproportion</td>
		<td style="width:50px">decimal</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Frequency as proportion.<br></td>
	</tr>
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column identifier should unique.</td></tr>
</table>
</p>


<h3 id="phenotypeProperty_entity">File: phenotypeproperty.txt</h3> 

<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">identifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>user supplied or automatically assigned (using a decorator) unique and short identifier, e.g. MA1234.<br></td>
	</tr>
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Phenotype Property.<br></td>
	</tr>
	<tr>
		<td style="width:150px">description</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>(Optional) Rudimentary meta data about the observable feature. Use of ontology       terms references to establish unambigious descriptions is recommended.<br></td>
	</tr>
	<tr>
		<td style="width:150px">unit_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>(Optional) Reference to the well-defined measurement unit used to observe this feature       (if feature is that concrete). E.g. mmHg.
		This xref uses {unit_identifier} to find related elements in file <a href="#ontologyTerm_entity">ontologyTerm.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">datatype</td>
		<td style="width:50px">enum</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">string</td>
		<td>(Optional) Reference to the technical data type. E.g. 'int'.<br></td>
	</tr>
	<tr>
		<td style="width:150px">temporal</td>
		<td style="width:50px">bool</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">false</td>
		<td>Whether this feature is time dependent and can have different values when measured       on different times (e.g. weight, temporal=true) or generally only measured once (e.g. birth date,       temporal=false).<br></td>
	</tr>
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column identifier should unique.</td></tr>
</table>
</p>


<h3 id="phenotypeMethod_entity">File: phenotypemethod.txt</h3> 

<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">identifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>user supplied or automatically assigned (using a decorator) unique and short identifier, e.g. MA1234.<br></td>
	</tr>
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Name.<br></td>
	</tr>
	<tr>
		<td style="width:150px">description</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>(Optional) Rudimentary meta data about the observable feature. Use of ontology       terms references to establish unambigious descriptions is recommended.<br></td>
	</tr>
	<tr>
		<td style="width:150px">protocolused_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Reference to the protocol that is being used (if available).
		This xref uses {protocolused_identifier} to find related elements in file <a href="#protocol_entity">protocol.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">starttime</td>
		<td style="width:50px">datetime</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">today</td>
		<td>time when the protocol started.<br></td>
	</tr>
	<tr>
		<td style="width:150px">endtime</td>
		<td style="width:50px">datetime</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">today</td>
		<td>(Optional) time when the protocol ended.<br></td>
	</tr>
	<tr>
		<td style="width:150px">studyid_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Points to study that this method came in.
		This xref uses {studyid_identifier} to find related elements in file <a href="#study_entity">study.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">phenotypepropertyid_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Phenotype property name.
		This xref uses {phenotypepropertyid_identifier} to find related elements in file <a href="#phenotypeProperty_entity">phenotypeProperty.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">sample</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Biological system or sample type being assessed.<br></td>
	</tr>
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column identifier should unique.</td></tr>
</table>
</p>


<h3 id="phenotypeValue_entity">File: phenotypevalue.txt</h3> 

<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">observationset_id<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Reference to the observation. For example a particular patient visit or the application of a microarray or the calculation of a QTL model.
		This xref uses {observationset_id} to find related elements in file <a href="#observationSet_entity">observationSet.txt</a> based on unique column {id}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">feature_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>References the ObservableFeature that this observation was made on. For example 'probe123'..
		This xref uses {feature_identifier} to find related elements in file <a href="#observableFeature_entity">observableFeature.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">characteristic_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Is brother of [characteristic], or Average of [height].
		This xref uses {characteristic_identifier} to find related elements in file <a href="#characteristic_entity">characteristic.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">value</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Value.<br></td>
	</tr>
	<tr>
		<td style="width:150px">identifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>user supplied or automatically assigned (using a decorator) unique and short identifier, e.g. MA1234.<br></td>
	</tr>
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Name.<br></td>
	</tr>
	<tr>
		<td style="width:150px">phenotypepropertyid_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Phenotype property name.
		This xref uses {phenotypepropertyid_identifier} to find related elements in file <a href="#phenotypeProperty_entity">phenotypeProperty.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">valuerank</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Value Rank.<br></td>
	</tr>
	<tr>
		<td style="width:150px">valueismean</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Value is mean.<br></td>
	</tr>
	<tr>
		<td style="width:150px">std</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Standard Deviation.<br></td>
	</tr>
	<tr>
		<td style="width:150px">min</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Minimum value.<br></td>
	</tr>
	<tr>
		<td style="width:150px">max</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Maximum value.<br></td>
	</tr>
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column identifier should unique.</td></tr>
</table>
</p>


<h3 id="samplePanel_entity">File: samplepanel.txt</h3> 

<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">identifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>user supplied or automatically assigned (using a decorator) unique and short identifier, e.g. MA1234.<br></td>
	</tr>
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Name.<br></td>
	</tr>
	<tr>
		<td style="width:150px">description</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Description.<br></td>
	</tr>
	<tr>
		<td style="width:150px">paneltype_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Indicate the type of Panel (example: Sample panel, AssayedPanel, Natural=wild type, Parental=parents of a cross, F1=First generation of cross, RCC=Recombinant congenic, CSS=chromosome substitution).
		This xref uses {paneltype_identifier} to find related elements in file <a href="#ontologyTerm_entity">ontologyTerm.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">numberofindividuals</td>
		<td style="width:50px">int</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>NumberOfIndividuals.<br></td>
	</tr>
	<tr>
		<td style="width:150px">species_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>The species this panel is an instance of/part of/extracted from..
		This xref uses {species_identifier} to find related elements in file <a href="#species_entity">species.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">individuals_Identifier<br></td>
		<td style="width:50px">mref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>The list of individuals in this panel.
		This mref uses {individuals_identifier} to find related elements in file <a href="#individual_entity">individual.txt</a> based on unique column {identifier}.
		. More than one reference can be added separated by '|', for example: ref1|ref2|ref3.</td>
	</tr>	
	<tr>
		<td style="width:150px">centralidentifier_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>The central GWAS identifier for this panel.
		This xref uses {centralidentifier_identifier} to find related elements in file <a href="#ontologyTerm_entity">ontologyTerm.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">composition</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Composition.<br></td>
	</tr>
	<tr>
		<td style="width:150px">totalnumberofindividuals</td>
		<td style="width:50px">int</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>*...but required for association datasets.<br></td>
	</tr>
	<tr>
		<td style="width:150px">numberofsexmale</td>
		<td style="width:50px">int</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Number of males.<br></td>
	</tr>
	<tr>
		<td style="width:150px">numberofsexfemale</td>
		<td style="width:50px">int</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Number of females.<br></td>
	</tr>
	<tr>
		<td style="width:150px">numberofsexunknown</td>
		<td style="width:50px">int</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Number of unknown sex.<br></td>
	</tr>
	<tr>
		<td style="width:150px">numberofprobands</td>
		<td style="width:50px">int</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>field used only if Composition = 'Trios'.<br></td>
	</tr>
	<tr>
		<td style="width:150px">numberofparents</td>
		<td style="width:50px">int</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>field used only if Composition = 'Trios'.<br></td>
	</tr>
	<tr>
		<td style="width:150px">modeofrecruitment</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Mode of recruitment.<br></td>
	</tr>
	<tr>
		<td style="width:150px">diagnosisagerange</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Diagnosis age range.<br></td>
	</tr>
	<tr>
		<td style="width:150px">diagnosisperiod</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Diagnosis period.<br></td>
	</tr>
	<tr>
		<td style="width:150px">samplingagerange</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Sampling age range.<br></td>
	</tr>
	<tr>
		<td style="width:150px">samplingperiod</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Sampling period.<br></td>
	</tr>
	<tr>
		<td style="width:150px">populationinfo</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Population information.<br></td>
	</tr>
	<tr>
		<td style="width:150px">geographicregioninfo</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Geographic region information.<br></td>
	</tr>
	<tr>
		<td style="width:150px">ethnicityinfo</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Ethnicity information.<br></td>
	</tr>
	<tr>
		<td style="width:150px">birthplaceinfo</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Birth place information.<br></td>
	</tr>
	<tr>
		<td style="width:150px">admixtureinfo</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Admixture information.<br></td>
	</tr>
	<tr>
		<td style="width:150px">environmentinfo</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Environment information.<br></td>
	</tr>
	<tr>
		<td style="width:150px">sourceofdna</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>In (SELECT [SourceOfDNA] FROM [Sampleset_SourceOfDNAList];).<br></td>
	</tr>
	<tr>
		<td style="width:150px">dnasarepooled</td>
		<td style="width:50px">enum</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">Undefined</td>
		<td>Are DNAs pooled?.<br></td>
	</tr>
	<tr>
		<td style="width:150px">dnasarewga</td>
		<td style="width:50px">enum</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">Undefined</td>
		<td>Are DNAs WGA?.<br></td>
	</tr>
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column identifier should unique.</td></tr>
</table>
</p>


<h3 id="assayedPanel_entity">File: assayedpanel.txt</h3> 

<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">identifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>user supplied or automatically assigned (using a decorator) unique and short identifier, e.g. MA1234.<br></td>
	</tr>
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Name.<br></td>
	</tr>
	<tr>
		<td style="width:150px">description</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Description.<br></td>
	</tr>
	<tr>
		<td style="width:150px">paneltype_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Indicate the type of Panel (example: Sample panel, AssayedPanel, Natural=wild type, Parental=parents of a cross, F1=First generation of cross, RCC=Recombinant congenic, CSS=chromosome substitution).
		This xref uses {paneltype_identifier} to find related elements in file <a href="#ontologyTerm_entity">ontologyTerm.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">numberofindividuals</td>
		<td style="width:50px">int</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>NumberOfIndividuals.<br></td>
	</tr>
	<tr>
		<td style="width:150px">species_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>The species this panel is an instance of/part of/extracted from..
		This xref uses {species_identifier} to find related elements in file <a href="#species_entity">species.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">individuals_Identifier<br></td>
		<td style="width:50px">mref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>The list of individuals in this panel.
		This mref uses {individuals_identifier} to find related elements in file <a href="#individual_entity">individual.txt</a> based on unique column {identifier}.
		. More than one reference can be added separated by '|', for example: ref1|ref2|ref3.</td>
	</tr>	
	<tr>
		<td style="width:150px">totalnumberofindividuals</td>
		<td style="width:50px">int</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>*...but required for association datasets.<br></td>
	</tr>
	<tr>
		<td style="width:150px">numberofsexmale</td>
		<td style="width:50px">int</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Number of males.<br></td>
	</tr>
	<tr>
		<td style="width:150px">numberofsexfemale</td>
		<td style="width:50px">int</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Number of females.<br></td>
	</tr>
	<tr>
		<td style="width:150px">numberofsexunknown</td>
		<td style="width:50px">int</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Number of unknown sex.<br></td>
	</tr>
	<tr>
		<td style="width:150px">numberofprobands</td>
		<td style="width:50px">int</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>field used only if Composition = 'Trios'.<br></td>
	</tr>
	<tr>
		<td style="width:150px">numberofparents</td>
		<td style="width:50px">int</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>field used only if Composition = 'Trios'.<br></td>
	</tr>
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column identifier should unique.</td></tr>
</table>
</p>


<h3 id="gWASExperiment_entity">File: gwasexperiment.txt</h3> 

<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">identifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>user supplied or automatically assigned (using a decorator) unique and short identifier, e.g. MA1234.<br></td>
	</tr>
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>human readible name, not necessary unique.<br></td>
	</tr>
	<tr>
		<td style="width:150px">study_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Part of Study..
		This xref uses {study_identifier} to find related elements in file <a href="#study_entity">study.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">design</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Design.<br></td>
	</tr>
	<tr>
		<td style="width:150px">experimenttype_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Experiment type. E.g. 'case-control'..
		This xref uses {experimenttype_identifier} to find related elements in file <a href="#ontologyTerm_entity">ontologyTerm.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">totalmarkerstested</td>
		<td style="width:50px">int</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Total markers tested.<br></td>
	</tr>
	<tr>
		<td style="width:150px">totalmarkersimported</td>
		<td style="width:50px">int</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Total markers imported.<br></td>
	</tr>
	<tr>
		<td style="width:150px">objective</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Objective.<br></td>
	</tr>
	<tr>
		<td style="width:150px">outcome</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Outcome.<br></td>
	</tr>
	<tr>
		<td style="width:150px">comments</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Comments.<br></td>
	</tr>
	<tr>
		<td style="width:150px">individualdatastatement</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">Access to individual-level data must be made to the study authors</td>
		<td>Individual data statement.<br></td>
	</tr>
	<tr>
		<td style="width:150px">timecreated</td>
		<td style="width:50px">datetime</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Time created.<br></td>
	</tr>
	<tr>
		<td style="width:150px">assayedpanels_Identifier<br></td>
		<td style="width:50px">mref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Assayed panels.
		This mref uses {assayedpanels_identifier} to find related elements in file <a href="#panel_entity">panel.txt</a> based on unique column {identifier}.
		. More than one reference can be added separated by '|', for example: ref1|ref2|ref3.</td>
	</tr>	
	<tr>
		<td style="width:150px">datasets_Identifier<br></td>
		<td style="width:50px">mref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>DataSets that were input/output of this experiment..
		This mref uses {datasets_identifier} to find related elements in file <a href="#dataSet_entity">dataSet.txt</a> based on unique column {identifier}.
		. More than one reference can be added separated by '|', for example: ref1|ref2|ref3.</td>
	</tr>	
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column identifier should unique.</td></tr>
</table>
</p>


<h3 id="usedMarkerSet_entity">File: usedmarkerset.txt</h3> 

<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">identifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>user supplied or automatically assigned (using a decorator) unique and short identifier, e.g. MA1234.<br></td>
	</tr>
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>human readible name, not necessary unique.<br></td>
	</tr>
	<tr>
		<td style="width:150px">description</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>(Optional) Rudimentary meta data about the observable feature. Use of ontology       terms references to establish unambigious descriptions is recommended.<br></td>
	</tr>
	<tr>
		<td style="width:150px">unit_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>(Optional) Reference to the well-defined measurement unit used to observe this feature       (if feature is that concrete). E.g. mmHg.
		This xref uses {unit_identifier} to find related elements in file <a href="#ontologyTerm_entity">ontologyTerm.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">datatype</td>
		<td style="width:50px">enum</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">string</td>
		<td>(Optional) Reference to the technical data type. E.g. 'int'.<br></td>
	</tr>
	<tr>
		<td style="width:150px">temporal</td>
		<td style="width:50px">bool</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">false</td>
		<td>Whether this feature is time dependent and can have different values when measured       on different times (e.g. weight, temporal=true) or generally only measured once (e.g. birth date,       temporal=false).<br></td>
	</tr>
	<tr>
		<td style="width:150px">experimentid_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Experiment ID.
		This xref uses {experimentid_identifier} to find related elements in file <a href="#experiment_entity">experiment.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">markeridentifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Marker identifier.<br></td>
	</tr>
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column identifier should unique.</td></tr>
<tr><td colspan="5">Contraint: values in the combined columns (experimentid, markeridentifier) should be unique.</td></tr>
</table>
</p>


<h3 id="significance_entity">File: significance.txt</h3> 

<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">identifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>user supplied or automatically assigned (using a decorator) unique and short identifier, e.g. MA1234.<br></td>
	</tr>
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>human readible name, not necessary unique.<br></td>
	</tr>
	<tr>
		<td style="width:150px">description</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>(Optional) Rudimentary meta data about the observable feature. Use of ontology       terms references to establish unambigious descriptions is recommended.<br></td>
	</tr>
	<tr>
		<td style="width:150px">protocolused_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Reference to the protocol that is being used (if available).
		This xref uses {protocolused_identifier} to find related elements in file <a href="#protocol_entity">protocol.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">starttime</td>
		<td style="width:50px">datetime</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">today</td>
		<td>time when the protocol started.<br></td>
	</tr>
	<tr>
		<td style="width:150px">endtime</td>
		<td style="width:50px">datetime</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">today</td>
		<td>(Optional) time when the protocol ended.<br></td>
	</tr>
	<tr>
		<td style="width:150px">usedmarkersetid_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Used marker set ID.
		This xref uses {usedmarkersetid_identifier} to find related elements in file <a href="#usedMarkerSet_entity">usedMarkerSet.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">neglogpvalue</td>
		<td style="width:50px">decimal</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Negative log p-value.<br></td>
	</tr>
	<tr>
		<td style="width:150px">unadjustedpvalue</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Unadjusted p-value.<br></td>
	</tr>
	<tr>
		<td style="width:150px">adjustedpvalue</td>
		<td style="width:50px">decimal</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Adjusted p-value.<br></td>
	</tr>
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column identifier should unique.</td></tr>
</table>
</p>


<h3 id="effectSize_entity">File: effectsize.txt</h3> 

<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">identifier</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>user supplied or automatically assigned (using a decorator) unique and short identifier, e.g. MA1234.<br></td>
	</tr>
	<tr>
		<td style="width:150px">name</td>
		<td style="width:50px">string</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>human readible name, not necessary unique.<br></td>
	</tr>
	<tr>
		<td style="width:150px">description</td>
		<td style="width:50px">text</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>(Optional) Rudimentary meta data about the observable feature. Use of ontology       terms references to establish unambigious descriptions is recommended.<br></td>
	</tr>
	<tr>
		<td style="width:150px">protocolused_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Reference to the protocol that is being used (if available).
		This xref uses {protocolused_identifier} to find related elements in file <a href="#protocol_entity">protocol.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">starttime</td>
		<td style="width:50px">datetime</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">today</td>
		<td>time when the protocol started.<br></td>
	</tr>
	<tr>
		<td style="width:150px">endtime</td>
		<td style="width:50px">datetime</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">today</td>
		<td>(Optional) time when the protocol ended.<br></td>
	</tr>
	<tr>
		<td style="width:150px">usedmarkersetid_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">&nbsp;</td>
		<td style="width:50px">&nbsp;</td>
		<td>Used Marker ID.
		This xref uses {usedmarkersetid_identifier} to find related elements in file <a href="#usedMarkerSet_entity">usedMarkerSet.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">lower95bound</td>
		<td style="width:50px">decimal</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Lower95Bound.<br></td>
	</tr>
	<tr>
		<td style="width:150px">upper95bound</td>
		<td style="width:50px">decimal</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Upper95Bound.<br></td>
	</tr>
	<tr>
		<td style="width:150px">stderror</td>
		<td style="width:50px">decimal</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>StdError.<br></td>
	</tr>
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Constraint: values in column identifier should unique.</td></tr>
</table>
</p>


<h3 id="selectionCriteria_entity">File: selectioncriteria.txt</h3> 
<p>Contents:<br> Create relationships between panels, founder panels,
				such as overlap,
				selection criteria, getting assayed panel from a
				sample panel, etc.
			
</p>
<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">sourcepanel_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>SourcePanel.
		This xref uses {sourcepanel_identifier} to find related elements in file <a href="#panel_entity">panel.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">targetpanel_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>TargetPanel.
		This xref uses {targetpanel_identifier} to find related elements in file <a href="#panel_entity">panel.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">numberofindividuals</td>
		<td style="width:50px">int</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>NumberOfIndividuals.<br></td>
	</tr>
	<tr>
		<td style="width:150px">details</td>
		<td style="width:50px">text</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>Details.<br></td>
	</tr>
	<!-- all optional fields -->
	<!-- all the uniques -->
</table>
</p>



<h1>Appendix: documentation of the mref tables</h1>
<h2>org.molgenis file types</h2>
<h3 id="protocol_subprotocols_entity">File: protocol_subprotocols.txt</h3> 
<p>Contents:<br> Link table for many-to-many relationship 'Protocol.subprotocols'.</p>
<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">subprotocols_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>
		This xref uses {subprotocols_identifier} to find related elements in file <a href="#protocol_entity">protocol.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">protocol_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>
		This xref uses {protocol_identifier} to find related elements in file <a href="#protocol_entity">protocol.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Contraint: values in the combined columns (subprotocols, protocol) should be unique.</td></tr>
</table>
</p>


<h3 id="protocol_Features_entity">File: protocol_features.txt</h3> 
<p>Contents:<br> Link table for many-to-many relationship 'Protocol.Features'.</p>
<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">features_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>
		This xref uses {features_identifier} to find related elements in file <a href="#observableFeature_entity">observableFeature.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">protocol_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>
		This xref uses {protocol_identifier} to find related elements in file <a href="#protocol_entity">protocol.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Contraint: values in the combined columns (features, protocol) should be unique.</td></tr>
</table>
</p>


<h3 id="panel_Individuals_entity">File: panel_individuals.txt</h3> 
<p>Contents:<br> Link table for many-to-many relationship 'Panel.Individuals'.</p>
<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">individuals_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>
		This xref uses {individuals_identifier} to find related elements in file <a href="#individual_entity">individual.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">panel_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>
		This xref uses {panel_identifier} to find related elements in file <a href="#panel_entity">panel.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Contraint: values in the combined columns (individuals, panel) should be unique.</td></tr>
</table>
</p>


<h3 id="experiment_AssayedPanels_entity">File: experiment_assayedpanels.txt</h3> 
<p>Contents:<br> Link table for many-to-many relationship 'Experiment.AssayedPanels'.</p>
<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">assayedpanels_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>
		This xref uses {assayedpanels_identifier} to find related elements in file <a href="#panel_entity">panel.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">experiment_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>
		This xref uses {experiment_identifier} to find related elements in file <a href="#experiment_entity">experiment.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Contraint: values in the combined columns (assayedpanels, experiment) should be unique.</td></tr>
</table>
</p>


<h3 id="experiment_DataSets_entity">File: experiment_datasets.txt</h3> 
<p>Contents:<br> Link table for many-to-many relationship 'Experiment.DataSets'.</p>
<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">datasets_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>
		This xref uses {datasets_identifier} to find related elements in file <a href="#dataSet_entity">dataSet.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">experiment_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>
		This xref uses {experiment_identifier} to find related elements in file <a href="#experiment_entity">experiment.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Contraint: values in the combined columns (datasets, experiment) should be unique.</td></tr>
</table>
</p>


<h3 id="person_AffiliateInstitutions_entity">File: person_affiliateinstitutions.txt</h3> 
<p>Contents:<br> Link table for many-to-many relationship 'Person.AffiliateInstitutions'.</p>
<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">affiliateinstitutions_name<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>
		This xref uses {affiliateinstitutions_name} to find related elements in file <a href="#institute_entity">institute.txt</a> based on unique column {name}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">person_Name<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>
		This xref uses {person_name} to find related elements in file <a href="#person_entity">person.txt</a> based on unique column {name}.
		</td>
	</tr>	
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Contraint: values in the combined columns (affiliateinstitutions, person) should be unique.</td></tr>
</table>
</p>


<h3 id="citation_ontologyTerms_entity">File: citation_ontologyterms.txt</h3> 
<p>Contents:<br> Link table for many-to-many relationship 'Citation.ontologyTerms'.</p>
<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">ontologyterms_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>
		This xref uses {ontologyterms_identifier} to find related elements in file <a href="#ontologyTerm_entity">ontologyTerm.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">citation_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>
		This xref uses {citation_identifier} to find related elements in file <a href="#citation_entity">citation.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Contraint: values in the combined columns (ontologyterms, citation) should be unique.</td></tr>
</table>
</p>


<h3 id="studyDetails_otherCitations_entity">File: studydetails_othercitations.txt</h3> 
<p>Contents:<br> Link table for many-to-many relationship 'StudyDetails.otherCitations'.</p>
<p>Structure:<br>
<table style="width:100%; cellspacing:0px; border:1px #000000;">
	<!-- table header -->
	<tr>
		<th>column name</th>
		<th>type</th>
		<th>required?</th>
		<th>auto/default</th>
		<th>description</th>
	</tr>
	
	<!-- all required fields -->
	<tr>
		<td style="width:150px">othercitations_Identifier<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>
		This xref uses {othercitations_identifier} to find related elements in file <a href="#citation_entity">citation.txt</a> based on unique column {identifier}.
		</td>
	</tr>	
	<tr>
		<td style="width:150px">studydetails_id<br></td>
		<td style="width:50px">xref</td>
		<td style="width:20px">YES</td>
		<td style="width:50px">&nbsp;</td>
		<td>
		This xref uses {studydetails_id} to find related elements in file <a href="#studyDetails_entity">studyDetails.txt</a> based on unique column {id}.
		</td>
	</tr>	
	<!-- all optional fields -->
	<!-- all the uniques -->
<tr><td colspan="5">Contraint: values in the combined columns (othercitations, studydetails) should be unique.</td></tr>
</table>
</p>



</body></html>

